#' Genomics API
#' Stores, processes, explores and shares genomic data.
#' 
#' Auto-generated code by googleAuthR::gar_create_api_skeleton
#'  at 2016-09-03 23:19:39
#' filename: /Users/mark/dev/R/autoGoogleAPI/googlegenomicsv1.auto/R/genomics_functions.R
#' api_json: api_json
#' 
#' @details 
#' Authentication scopes used are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/bigquery
#' \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' @docType package 
#' @name genomics_googleAuthR
#' 
NULL
## NULL


#' Creates a new annotation set. Caller must have WRITE permission for the associated dataset. The following fields are required: * datasetId * referenceSetId All other fields may be optionally specified, unless documented as being server-generated (for example, the `id` field).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param AnnotationSet The \link{AnnotationSet} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family AnnotationSet functions
#' @export
annotationsets.create <- function(AnnotationSet) {
    url <- "https://genomics.googleapis.com/v1/annotationsets"
    # genomics.annotationsets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(AnnotationSet, "gar_AnnotationSet"))
    
    f(the_body = AnnotationSet)
    
}

#' Gets an annotation set. Caller must have READ permission for the associated dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationSetId The ID of the annotation set to be retrieved
#' @importFrom googleAuthR gar_api_generator
#' @export
annotationsets.get <- function(annotationSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/annotationsets/%s", annotationSetId)
    # genomics.annotationsets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Updates an annotation set. The update must respect all mutability restrictions and other invariants described on the annotation set resource. Caller must have WRITE permission for the associated dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param AnnotationSet The \link{AnnotationSet} object to pass to this method
#' @param annotationSetId The ID of the annotation set to be updated
#' @param updateMask An optional mask specifying which fields to update
#' @importFrom googleAuthR gar_api_generator
#' @family AnnotationSet functions
#' @export
annotationsets.update <- function(AnnotationSet, annotationSetId, updateMask = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/annotationsets/%s", annotationSetId)
    # genomics.annotationsets.update
    f <- gar_api_generator(url, "PUT", pars_args = list(updateMask = updateMask), 
        data_parse_function = function(x) x)
    stopifnot(inherits(AnnotationSet, "gar_AnnotationSet"))
    
    f(the_body = AnnotationSet)
    
}

#' Deletes an annotation set. Caller must have WRITE permission for the associated annotation set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationSetId The ID of the annotation set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
annotationsets.delete <- function(annotationSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/annotationsets/%s", annotationSetId)
    # genomics.annotationsets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Searches for annotation sets that match the given criteria. Annotation sets are returned in an unspecified order. This order is consistent, such that two queries for the same content (regardless of page size) yield annotation sets in the same order across their respective streams of paginated responses. Caller must have READ permission for the queried datasets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchAnnotationSetsRequest The \link{SearchAnnotationSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchAnnotationSetsRequest functions
#' @export
annotationsets.search <- function(SearchAnnotationSetsRequest) {
    url <- "https://genomics.googleapis.com/v1/annotationsets/search"
    # genomics.annotationsets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchAnnotationSetsRequest, "gar_SearchAnnotationSetsRequest"))
    
    f(the_body = SearchAnnotationSetsRequest)
    
}

#' Creates a new annotation. Caller must have WRITE permission for the associated annotation set. The following fields are required: * annotationSetId * referenceName or referenceId ### Transcripts For annotations of type TRANSCRIPT, the following fields of transcript must be provided: * exons.start * exons.end All other fields may be optionally specified, unless documented as being server-generated (for example, the `id` field). The annotated range must be no longer than 100Mbp (mega base pairs). See the Annotation resource for additional restrictions on each field.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Annotation The \link{Annotation} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family Annotation functions
#' @export
annotations.create <- function(Annotation) {
    url <- "https://genomics.googleapis.com/v1/annotations"
    # genomics.annotations.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(Annotation, "gar_Annotation"))
    
    f(the_body = Annotation)
    
}

#' Creates one or more new annotations atomically. All annotations must belong to the same annotation set. Caller must have WRITE permission for this annotation set. For optimal performance, batch positionally adjacent annotations together. If the request has a systemic issue, such as an attempt to write to an inaccessible annotation set, the entire RPC will fail accordingly. For lesser data issues, when possible an error will be isolated to the corresponding batch entry in the response; the remaining well formed annotations will be created normally. For details on the requirements for each individual annotation resource, see CreateAnnotation.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param BatchCreateAnnotationsRequest The \link{BatchCreateAnnotationsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family BatchCreateAnnotationsRequest functions
#' @export
annotations.batchCreate <- function(BatchCreateAnnotationsRequest) {
    url <- "https://genomics.googleapis.com/v1/annotations:batchCreate"
    # genomics.annotations.batchCreate
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(BatchCreateAnnotationsRequest, "gar_BatchCreateAnnotationsRequest"))
    
    f(the_body = BatchCreateAnnotationsRequest)
    
}

#' Gets an annotation. Caller must have READ permission for the associated annotation set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationId The ID of the annotation to be retrieved
#' @importFrom googleAuthR gar_api_generator
#' @export
annotations.get <- function(annotationId) {
    url <- sprintf("https://genomics.googleapis.com/v1/annotations/%s", annotationId)
    # genomics.annotations.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Updates an annotation. Caller must have WRITE permission for the associated dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Annotation The \link{Annotation} object to pass to this method
#' @param annotationId The ID of the annotation to be updated
#' @param updateMask An optional mask specifying which fields to update
#' @importFrom googleAuthR gar_api_generator
#' @family Annotation functions
#' @export
annotations.update <- function(Annotation, annotationId, updateMask = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/annotations/%s", annotationId)
    # genomics.annotations.update
    f <- gar_api_generator(url, "PUT", pars_args = list(updateMask = updateMask), 
        data_parse_function = function(x) x)
    stopifnot(inherits(Annotation, "gar_Annotation"))
    
    f(the_body = Annotation)
    
}

#' Deletes an annotation. Caller must have WRITE permission for the associated annotation set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationId The ID of the annotation to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
annotations.delete <- function(annotationId) {
    url <- sprintf("https://genomics.googleapis.com/v1/annotations/%s", annotationId)
    # genomics.annotations.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Searches for annotations that match the given criteria. Results are ordered by genomic coordinate (by reference sequence, then position). Annotations with equivalent genomic coordinates are returned in an unspecified order. This order is consistent, such that two queries for the same content (regardless of page size) yield annotations in the same order across their respective streams of paginated responses. Caller must have READ permission for the queried annotation sets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchAnnotationsRequest The \link{SearchAnnotationsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchAnnotationsRequest functions
#' @export
annotations.search <- function(SearchAnnotationsRequest) {
    url <- "https://genomics.googleapis.com/v1/annotations/search"
    # genomics.annotations.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchAnnotationsRequest, "gar_SearchAnnotationsRequest"))
    
    f(the_body = SearchAnnotationsRequest)
    
}

#' Lists datasets within a project. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param projectId Required
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @importFrom googleAuthR gar_api_generator
#' @export
datasets.list <- function(projectId = NULL, pageSize = NULL, pageToken = NULL) {
    url <- "https://genomics.googleapis.com/v1/datasets"
    # genomics.datasets.list
    f <- gar_api_generator(url, "GET", pars_args = list(projectId = projectId, pageSize = pageSize, 
        pageToken = pageToken), data_parse_function = function(x) x)
    f()
    
}

#' Creates a new dataset. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Dataset The \link{Dataset} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family Dataset functions
#' @export
datasets.create <- function(Dataset) {
    url <- "https://genomics.googleapis.com/v1/datasets"
    # genomics.datasets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(Dataset, "gar_Dataset"))
    
    f(the_body = Dataset)
    
}

#' Gets a dataset by ID. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param datasetId The ID of the dataset
#' @importFrom googleAuthR gar_api_generator
#' @export
datasets.get <- function(datasetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/datasets/%s", datasetId)
    # genomics.datasets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Updates a dataset. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Dataset The \link{Dataset} object to pass to this method
#' @param datasetId The ID of the dataset to be updated
#' @param updateMask An optional mask specifying which fields to update
#' @importFrom googleAuthR gar_api_generator
#' @family Dataset functions
#' @export
datasets.patch <- function(Dataset, datasetId, updateMask = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/datasets/%s", datasetId)
    # genomics.datasets.patch
    f <- gar_api_generator(url, "PATCH", pars_args = list(updateMask = updateMask), 
        data_parse_function = function(x) x)
    stopifnot(inherits(Dataset, "gar_Dataset"))
    
    f(the_body = Dataset)
    
}

#' Deletes a dataset and all of its contents (all read group sets, reference sets, variant sets, call sets, annotation sets, etc.) This is reversible (up to one week after the deletion) via the datasets.undelete operation. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param datasetId The ID of the dataset to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
datasets.delete <- function(datasetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/datasets/%s", datasetId)
    # genomics.datasets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Undeletes a dataset by restoring a dataset which was deleted via this API. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) This operation is only possible for a week after the deletion occurred.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param UndeleteDatasetRequest The \link{UndeleteDatasetRequest} object to pass to this method
#' @param datasetId The ID of the dataset to be undeleted
#' @importFrom googleAuthR gar_api_generator
#' @family UndeleteDatasetRequest functions
#' @export
datasets.undelete <- function(UndeleteDatasetRequest, datasetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/datasets/%s:undelete", datasetId)
    # genomics.datasets.undelete
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(UndeleteDatasetRequest, "gar_UndeleteDatasetRequest"))
    
    f(the_body = UndeleteDatasetRequest)
    
}

#' Sets the access control policy on the specified dataset. Replaces any existing policy. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) See Setting a Policy for more information.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SetIamPolicyRequest The \link{SetIamPolicyRequest} object to pass to this method
#' @param resource REQUIRED: The resource for which policy is being specified
#' @importFrom googleAuthR gar_api_generator
#' @family SetIamPolicyRequest functions
#' @export
datasets.setIamPolicy <- function(SetIamPolicyRequest, resource) {
    url <- sprintf("https://genomics.googleapis.com/v1/{+resource}:setIamPolicy", 
        resource)
    # genomics.datasets.setIamPolicy
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SetIamPolicyRequest, "gar_SetIamPolicyRequest"))
    
    f(the_body = SetIamPolicyRequest)
    
}

#' Gets the access control policy for the dataset. This is empty if the policy or resource does not exist. See Getting a Policy for more information. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param GetIamPolicyRequest The \link{GetIamPolicyRequest} object to pass to this method
#' @param resource REQUIRED: The resource for which policy is being specified
#' @importFrom googleAuthR gar_api_generator
#' @family GetIamPolicyRequest functions
#' @export
datasets.getIamPolicy <- function(GetIamPolicyRequest, resource) {
    url <- sprintf("https://genomics.googleapis.com/v1/{+resource}:getIamPolicy", 
        resource)
    # genomics.datasets.getIamPolicy
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(GetIamPolicyRequest, "gar_GetIamPolicyRequest"))
    
    f(the_body = GetIamPolicyRequest)
    
}

#' Returns permissions that a caller has on the specified resource. See Testing Permissions for more information. For the definitions of datasets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param TestIamPermissionsRequest The \link{TestIamPermissionsRequest} object to pass to this method
#' @param resource REQUIRED: The resource for which policy is being specified
#' @importFrom googleAuthR gar_api_generator
#' @family TestIamPermissionsRequest functions
#' @export
datasets.testIamPermissions <- function(TestIamPermissionsRequest, resource) {
    url <- sprintf("https://genomics.googleapis.com/v1/{+resource}:testIamPermissions", 
        resource)
    # genomics.datasets.testIamPermissions
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(TestIamPermissionsRequest, "gar_TestIamPermissionsRequest"))
    
    f(the_body = TestIamPermissionsRequest)
    
}

#' Gets the latest state of a long-running operation. Clients can use this method to poll the operation result at intervals as recommended by the API service.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param name The name of the operation resource
#' @importFrom googleAuthR gar_api_generator
#' @export
operations.get <- function(name) {
    url <- sprintf("https://genomics.googleapis.com/v1/{+name}", name)
    # genomics.operations.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Lists operations that match the specified filter in the request.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param name The name of the operation collection
#' @param filter A string for filtering Operations
#' @param pageSize The maximum number of results to return
#' @param pageToken The standard list page token
#' @importFrom googleAuthR gar_api_generator
#' @export
operations.list <- function(name, filter = NULL, pageSize = NULL, pageToken = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/{+name}", name)
    # genomics.operations.list
    f <- gar_api_generator(url, "GET", pars_args = list(filter = filter, pageSize = pageSize, 
        pageToken = pageToken), data_parse_function = function(x) x)
    f()
    
}

#' Starts asynchronous cancellation on a long-running operation. The server makes a best effort to cancel the operation, but success is not guaranteed. Clients may use Operations.GetOperation or Operations.ListOperations to check whether the cancellation succeeded or the operation completed despite cancellation.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param CancelOperationRequest The \link{CancelOperationRequest} object to pass to this method
#' @param name The name of the operation resource to be cancelled
#' @importFrom googleAuthR gar_api_generator
#' @family CancelOperationRequest functions
#' @export
operations.cancel <- function(CancelOperationRequest, name) {
    url <- sprintf("https://genomics.googleapis.com/v1/{+name}:cancel", name)
    # genomics.operations.cancel
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(CancelOperationRequest, "gar_CancelOperationRequest"))
    
    f(the_body = CancelOperationRequest)
    
}

#' Creates read group sets by asynchronously importing the provided information. For the definitions of read group sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) The caller must have WRITE permissions to the dataset. ## Notes on [BAM](https://samtools.github.io/hts-specs/SAMv1.pdf) import - Tags will be converted to strings - tag types are not preserved - Comments (`@CO`) in the input file header will not be preserved - Original header order of references (`@SQ`) will not be preserved - Any reverse stranded unmapped reads will be reverse complemented, and their qualities (also the 'BQ' and 'OQ' tags, if any) will be reversed - Unmapped reads will be stripped of positional information (reference name and position)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/devstorage.read_write, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ImportReadGroupSetsRequest The \link{ImportReadGroupSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family ImportReadGroupSetsRequest functions
#' @export
readgroupsets.import <- function(ImportReadGroupSetsRequest) {
    url <- "https://genomics.googleapis.com/v1/readgroupsets:import"
    # genomics.readgroupsets.import
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ImportReadGroupSetsRequest, "gar_ImportReadGroupSetsRequest"))
    
    f(the_body = ImportReadGroupSetsRequest)
    
}

#' Exports a read group set to a BAM file in Google Cloud Storage. For the definitions of read group sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Note that currently there may be some differences between exported BAM files and the original BAM file at the time of import. See ImportReadGroupSets for caveats.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/devstorage.read_write, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ExportReadGroupSetRequest The \link{ExportReadGroupSetRequest} object to pass to this method
#' @param readGroupSetId Required
#' @importFrom googleAuthR gar_api_generator
#' @family ExportReadGroupSetRequest functions
#' @export
readgroupsets.export <- function(ExportReadGroupSetRequest, readGroupSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/readgroupsets/%s:export", 
        readGroupSetId)
    # genomics.readgroupsets.export
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ExportReadGroupSetRequest, "gar_ExportReadGroupSetRequest"))
    
    f(the_body = ExportReadGroupSetRequest)
    
}

#' Searches for read group sets matching the criteria. For the definitions of read group sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.searchReadGroupSets](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/readmethods.avdl#L135).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReadGroupSetsRequest The \link{SearchReadGroupSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReadGroupSetsRequest functions
#' @export
readgroupsets.search <- function(SearchReadGroupSetsRequest) {
    url <- "https://genomics.googleapis.com/v1/readgroupsets/search"
    # genomics.readgroupsets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReadGroupSetsRequest, "gar_SearchReadGroupSetsRequest"))
    
    f(the_body = SearchReadGroupSetsRequest)
    
}

#' Updates a read group set. For the definitions of read group sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ReadGroupSet The \link{ReadGroupSet} object to pass to this method
#' @param readGroupSetId The ID of the read group set to be updated
#' @param updateMask An optional mask specifying which fields to update
#' @importFrom googleAuthR gar_api_generator
#' @family ReadGroupSet functions
#' @export
readgroupsets.patch <- function(ReadGroupSet, readGroupSetId, updateMask = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/readgroupsets/%s", readGroupSetId)
    # genomics.readgroupsets.patch
    f <- gar_api_generator(url, "PATCH", pars_args = list(updateMask = updateMask), 
        data_parse_function = function(x) x)
    stopifnot(inherits(ReadGroupSet, "gar_ReadGroupSet"))
    
    f(the_body = ReadGroupSet)
    
}

#' Deletes a read group set. For the definitions of read group sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param readGroupSetId The ID of the read group set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
readgroupsets.delete <- function(readGroupSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/readgroupsets/%s", readGroupSetId)
    # genomics.readgroupsets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Gets a read group set by ID. For the definitions of read group sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param readGroupSetId The ID of the read group set
#' @importFrom googleAuthR gar_api_generator
#' @export
readgroupsets.get <- function(readGroupSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/readgroupsets/%s", readGroupSetId)
    # genomics.readgroupsets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Gets a list of reads for one or more read group sets. For the definitions of read group sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Reads search operates over a genomic coordinate space of reference sequence & position defined over the reference sequences to which the requested read group sets are aligned. If a target positional range is specified, search returns all reads whose alignment to the reference genome overlap the range. A query which specifies only read group set IDs yields all reads in those read group sets, including unmapped reads. All reads returned (including reads on subsequent pages) are ordered by genomic coordinate (by reference sequence, then position). Reads with equivalent genomic coordinates are returned in an unspecified order. This order is consistent, such that two queries for the same content (regardless of page size) yield reads in the same order across their respective streams of paginated responses. Implements [GlobalAllianceApi.searchReads](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/readmethods.avdl#L85).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReadsRequest The \link{SearchReadsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReadsRequest functions
#' @export
reads.search <- function(SearchReadsRequest) {
    url <- "https://genomics.googleapis.com/v1/reads/search"
    # genomics.reads.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReadsRequest, "gar_SearchReadsRequest"))
    
    f(the_body = SearchReadsRequest)
    
}

#' Returns a stream of all the reads matching the search request, ordered by reference name, position, and ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param StreamReadsRequest The \link{StreamReadsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family StreamReadsRequest functions
#' @export
reads.stream <- function(StreamReadsRequest) {
    url <- "https://genomics.googleapis.com/v1/reads:stream"
    # genomics.reads.stream
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(StreamReadsRequest, "gar_StreamReadsRequest"))
    
    f(the_body = StreamReadsRequest)
    
}

#' Searches for reference sets which match the given criteria. For the definitions of references and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.searchReferenceSets](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/referencemethods.avdl#L71)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReferenceSetsRequest The \link{SearchReferenceSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReferenceSetsRequest functions
#' @export
referencesets.search <- function(SearchReferenceSetsRequest) {
    url <- "https://genomics.googleapis.com/v1/referencesets/search"
    # genomics.referencesets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReferenceSetsRequest, "gar_SearchReferenceSetsRequest"))
    
    f(the_body = SearchReferenceSetsRequest)
    
}

#' Gets a reference set. For the definitions of references and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.getReferenceSet](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/referencemethods.avdl#L83).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param referenceSetId The ID of the reference set
#' @importFrom googleAuthR gar_api_generator
#' @export
referencesets.get <- function(referenceSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/referencesets/%s", referenceSetId)
    # genomics.referencesets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Searches for references which match the given criteria. For the definitions of references and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.searchReferences](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/referencemethods.avdl#L146).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReferencesRequest The \link{SearchReferencesRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReferencesRequest functions
#' @export
references.search <- function(SearchReferencesRequest) {
    url <- "https://genomics.googleapis.com/v1/references/search"
    # genomics.references.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReferencesRequest, "gar_SearchReferencesRequest"))
    
    f(the_body = SearchReferencesRequest)
    
}

#' Gets a reference. For the definitions of references and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.getReference](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/referencemethods.avdl#L158).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param referenceId The ID of the reference
#' @importFrom googleAuthR gar_api_generator
#' @export
references.get <- function(referenceId) {
    url <- sprintf("https://genomics.googleapis.com/v1/references/%s", referenceId)
    # genomics.references.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Creates variant data by asynchronously importing the provided information. For the definitions of variant sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) The variants for import will be merged with any existing variant that matches its reference sequence, start, end, reference bases, and alternative bases. If no such variant exists, a new one will be created. When variants are merged, the call information from the new variant is added to the existing variant, and Variant info fields are merged as specified in infoMergeConfig. As a special case, for single-sample VCF files, QUAL and FILTER fields will be moved to the call level; these are sometimes interpreted in a call-specific context. Imported VCF headers are appended to the metadata already in a variant set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/devstorage.read_write, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ImportVariantsRequest The \link{ImportVariantsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family ImportVariantsRequest functions
#' @export
variants.import <- function(ImportVariantsRequest) {
    url <- "https://genomics.googleapis.com/v1/variants:import"
    # genomics.variants.import
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ImportVariantsRequest, "gar_ImportVariantsRequest"))
    
    f(the_body = ImportVariantsRequest)
    
}

#' Gets a list of variants matching the criteria. For the definitions of variants and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.searchVariants](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/variantmethods.avdl#L126).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchVariantsRequest The \link{SearchVariantsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchVariantsRequest functions
#' @export
variants.search <- function(SearchVariantsRequest) {
    url <- "https://genomics.googleapis.com/v1/variants/search"
    # genomics.variants.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchVariantsRequest, "gar_SearchVariantsRequest"))
    
    f(the_body = SearchVariantsRequest)
    
}

#' Creates a new variant. For the definitions of variants and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Variant The \link{Variant} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family Variant functions
#' @export
variants.create <- function(Variant) {
    url <- "https://genomics.googleapis.com/v1/variants"
    # genomics.variants.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(Variant, "gar_Variant"))
    
    f(the_body = Variant)
    
}

#' Updates a variant. For the definitions of variants and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) This method supports patch semantics. Returns the modified variant without its calls.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Variant The \link{Variant} object to pass to this method
#' @param variantId The ID of the variant to be updated
#' @param updateMask An optional mask specifying which fields to update
#' @importFrom googleAuthR gar_api_generator
#' @family Variant functions
#' @export
variants.patch <- function(Variant, variantId, updateMask = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/variants/%s", variantId)
    # genomics.variants.patch
    f <- gar_api_generator(url, "PATCH", pars_args = list(updateMask = updateMask), 
        data_parse_function = function(x) x)
    stopifnot(inherits(Variant, "gar_Variant"))
    
    f(the_body = Variant)
    
}

#' Deletes a variant. For the definitions of variants and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantId The ID of the variant to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
variants.delete <- function(variantId) {
    url <- sprintf("https://genomics.googleapis.com/v1/variants/%s", variantId)
    # genomics.variants.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Gets a variant by ID. For the definitions of variants and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantId The ID of the variant
#' @importFrom googleAuthR gar_api_generator
#' @export
variants.get <- function(variantId) {
    url <- sprintf("https://genomics.googleapis.com/v1/variants/%s", variantId)
    # genomics.variants.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Merges the given variants with existing variants. For the definitions of variants and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Each variant will be merged with an existing variant that matches its reference sequence, start, end, reference bases, and alternative bases. If no such variant exists, a new one will be created. When variants are merged, the call information from the new variant is added to the existing variant. Variant info fields are merged as specified in the infoMergeConfig field of the MergeVariantsRequest. Please exercise caution when using this method! It is easy to introduce mistakes in existing variants and difficult to back out of them. For example, suppose you were trying to merge a new variant with an existing one and both variants contain calls that belong to callsets with the same callset ID. // Existing variant - irrelevant fields trimmed for clarity { 'variantSetId': '10473108253681171589', 'referenceName': '1', 'start': '10582', 'referenceBases': 'G', 'alternateBases': [ 'A' ], 'calls': [ { 'callSetId': '10473108253681171589-0', 'callSetName': 'CALLSET0', 'genotype': [ 0, 1 ], } ] } // New variant with conflicting call information { 'variantSetId': '10473108253681171589', 'referenceName': '1', 'start': '10582', 'referenceBases': 'G', 'alternateBases': [ 'A' ], 'calls': [ { 'callSetId': '10473108253681171589-0', 'callSetName': 'CALLSET0', 'genotype': [ 1, 1 ], } ] } The resulting merged variant would overwrite the existing calls with those from the new variant: { 'variantSetId': '10473108253681171589', 'referenceName': '1', 'start': '10582', 'referenceBases': 'G', 'alternateBases': [ 'A' ], 'calls': [ { 'callSetId': '10473108253681171589-0', 'callSetName': 'CALLSET0', 'genotype': [ 1, 1 ], } ] } This may be the desired outcome, but it is up to the user to determine if if that is indeed the case.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param MergeVariantsRequest The \link{MergeVariantsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family MergeVariantsRequest functions
#' @export
variants.merge <- function(MergeVariantsRequest) {
    url <- "https://genomics.googleapis.com/v1/variants:merge"
    # genomics.variants.merge
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(MergeVariantsRequest, "gar_MergeVariantsRequest"))
    
    f(the_body = MergeVariantsRequest)
    
}

#' Returns a stream of all the variants matching the search request, ordered by reference name, position, and ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param StreamVariantsRequest The \link{StreamVariantsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family StreamVariantsRequest functions
#' @export
variants.stream <- function(StreamVariantsRequest) {
    url <- "https://genomics.googleapis.com/v1/variants:stream"
    # genomics.variants.stream
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(StreamVariantsRequest, "gar_StreamVariantsRequest"))
    
    f(the_body = StreamVariantsRequest)
    
}

#' Creates a new variant set. For the definitions of variant sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) The provided variant set must have a valid `datasetId` set - all other fields are optional. Note that the `id` field will be ignored, as this is assigned by the server.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param VariantSet The \link{VariantSet} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family VariantSet functions
#' @export
variantsets.create <- function(VariantSet) {
    url <- "https://genomics.googleapis.com/v1/variantsets"
    # genomics.variantsets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(VariantSet, "gar_VariantSet"))
    
    f(the_body = VariantSet)
    
}

#' Exports variant set data to an external destination. For the definitions of variant sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/bigquery
#' \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/bigquery, https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ExportVariantSetRequest The \link{ExportVariantSetRequest} object to pass to this method
#' @param variantSetId Required
#' @importFrom googleAuthR gar_api_generator
#' @family ExportVariantSetRequest functions
#' @export
variantsets.export <- function(ExportVariantSetRequest, variantSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/variantsets/%s:export", variantSetId)
    # genomics.variantsets.export
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ExportVariantSetRequest, "gar_ExportVariantSetRequest"))
    
    f(the_body = ExportVariantSetRequest)
    
}

#' Gets a variant set by ID. For the definitions of variant sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantSetId Required
#' @importFrom googleAuthR gar_api_generator
#' @export
variantsets.get <- function(variantSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/variantsets/%s", variantSetId)
    # genomics.variantsets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Returns a list of all variant sets matching search criteria. For the definitions of variant sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.searchVariantSets](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/variantmethods.avdl#L49).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchVariantSetsRequest The \link{SearchVariantSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchVariantSetsRequest functions
#' @export
variantsets.search <- function(SearchVariantSetsRequest) {
    url <- "https://genomics.googleapis.com/v1/variantsets/search"
    # genomics.variantsets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchVariantSetsRequest, "gar_SearchVariantSetsRequest"))
    
    f(the_body = SearchVariantSetsRequest)
    
}

#' Deletes a variant set including all variants, call sets, and calls within. This is not reversible. For the definitions of variant sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantSetId The ID of the variant set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
variantsets.delete <- function(variantSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/variantsets/%s", variantSetId)
    # genomics.variantsets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Updates a variant set using patch semantics. For the definitions of variant sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param VariantSet The \link{VariantSet} object to pass to this method
#' @param variantSetId The ID of the variant to be updated (must already exist)
#' @param updateMask An optional mask specifying which fields to update
#' @importFrom googleAuthR gar_api_generator
#' @family VariantSet functions
#' @export
variantsets.patch <- function(VariantSet, variantSetId, updateMask = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/variantsets/%s", variantSetId)
    # genomics.variantsets.patch
    f <- gar_api_generator(url, "PATCH", pars_args = list(updateMask = updateMask), 
        data_parse_function = function(x) x)
    stopifnot(inherits(VariantSet, "gar_VariantSet"))
    
    f(the_body = VariantSet)
    
}

#' Gets a list of call sets matching the criteria. For the definitions of call sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Implements [GlobalAllianceApi.searchCallSets](https://github.com/ga4gh/schemas/blob/v0.5.1/src/main/resources/avro/variantmethods.avdl#L178).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchCallSetsRequest The \link{SearchCallSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchCallSetsRequest functions
#' @export
callsets.search <- function(SearchCallSetsRequest) {
    url <- "https://genomics.googleapis.com/v1/callsets/search"
    # genomics.callsets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchCallSetsRequest, "gar_SearchCallSetsRequest"))
    
    f(the_body = SearchCallSetsRequest)
    
}

#' Creates a new call set. For the definitions of call sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param CallSet The \link{CallSet} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family CallSet functions
#' @export
callsets.create <- function(CallSet) {
    url <- "https://genomics.googleapis.com/v1/callsets"
    # genomics.callsets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(CallSet, "gar_CallSet"))
    
    f(the_body = CallSet)
    
}

#' Updates a call set. For the definitions of call sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param CallSet The \link{CallSet} object to pass to this method
#' @param callSetId The ID of the call set to be updated
#' @param updateMask An optional mask specifying which fields to update
#' @importFrom googleAuthR gar_api_generator
#' @family CallSet functions
#' @export
callsets.patch <- function(CallSet, callSetId, updateMask = NULL) {
    url <- sprintf("https://genomics.googleapis.com/v1/callsets/%s", callSetId)
    # genomics.callsets.patch
    f <- gar_api_generator(url, "PATCH", pars_args = list(updateMask = updateMask), 
        data_parse_function = function(x) x)
    stopifnot(inherits(CallSet, "gar_CallSet"))
    
    f(the_body = CallSet)
    
}

#' Deletes a call set. For the definitions of call sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param callSetId The ID of the call set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
callsets.delete <- function(callSetId) {
    url <- sprintf("https://genomics.googleapis.com/v1/callsets/%s", callSetId)
    # genomics.callsets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}


#' Gets a call set by ID. For the definitions of call sets and other genomics resources, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://cloud.google.com/genomics/}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param callSetId The ID of the call set
#' @importFrom googleAuthR gar_api_generator
#' @export


callsets.get <- function(callSetId) {
    
    
    url <- sprintf("https://genomics.googleapis.com/v1/callsets/%s", callSetId)
    # genomics.callsets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    
    f()
    
    
}



