#' Genomics API Objects 
#' Stores, processes, explores and shares genomic data.
#' 
#' Auto-generated code by googleAuthR::gar_create_api_objects
#'  at 2016-09-03 23:19:40
#' filename: /Users/mark/dev/R/autoGoogleAPI/googlegenomicsv1.auto/R/genomics_objects.R
#' api_json: api_json
#' 
#' Objects for use by the functions created by googleAuthR::gar_create_api_skeleton

#' AnnotationSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An annotation set is a logical grouping of annotations that share consistent type information and provenance. Examples of annotation sets include 'all genes from refseq', and 'all variant annotations from ClinVar'.
#' 
#' @param AnnotationSet.info The \link{AnnotationSet.info} object or list of objects
#' @param id The server-generated annotation set ID, unique across all annotation sets
#' @param datasetId The dataset to which this annotation set belongs
#' @param referenceSetId The ID of the reference set that defines the coordinate space for this set's annotations
#' @param name The display name for this annotation set
#' @param sourceUri The source URI describing the file from which this annotation set was generated, if any
#' @param type The type of annotations contained within this set
#' @param info A map of additional read alignment information
#' 
#' @return AnnotationSet object
#' 
#' @family AnnotationSet functions
#' @export
AnnotationSet <- function(AnnotationSet.info = NULL, id = NULL, datasetId = NULL, 
    referenceSetId = NULL, name = NULL, sourceUri = NULL, type = NULL, info = NULL) {
    structure(list(AnnotationSet.info = AnnotationSet.info, id = id, datasetId = datasetId, 
        referenceSetId = referenceSetId, name = name, sourceUri = sourceUri, type = type, 
        info = info), class = "gar_AnnotationSet")
}

#' AnnotationSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read alignment information. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return AnnotationSet.info object
#' 
#' @family AnnotationSet functions
#' @export
AnnotationSet.info <- function() {
    list()
}

#' Empty Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A generic empty message that you can re-use to avoid defining duplicated empty messages in your APIs. A typical example is to use it as the request or the response type of an API method. For instance: service Foo { rpc Bar(google.protobuf.Empty) returns (google.protobuf.Empty); } The JSON representation for `Empty` is empty JSON object `{}`.
#' 
#' 
#' 
#' @return Empty object
#' 
#' @family Empty functions
#' @export
Empty <- function() {
    list()
}

#' SearchAnnotationSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param datasetIds Required
#' @param referenceSetId If specified, only annotation sets associated with the given reference set are returned
#' @param name Only return annotations sets for which a substring of the name matches this string (case insensitive)
#' @param types If specified, only annotation sets that have any of these types are returned
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchAnnotationSetsRequest object
#' 
#' @family SearchAnnotationSetsRequest functions
#' @export
SearchAnnotationSetsRequest <- function(datasetIds = NULL, referenceSetId = NULL, 
    name = NULL, types = NULL, pageToken = NULL, pageSize = NULL) {
    structure(list(datasetIds = datasetIds, referenceSetId = referenceSetId, name = name, 
        types = types, pageToken = pageToken, pageSize = pageSize), class = "gar_SearchAnnotationSetsRequest")
}

#' SearchAnnotationSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotationSets The matching annotation sets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchAnnotationSetsResponse object
#' 
#' @family SearchAnnotationSetsResponse functions
#' @export
SearchAnnotationSetsResponse <- function(annotationSets = NULL, nextPageToken = NULL) {
    structure(list(annotationSets = annotationSets, nextPageToken = nextPageToken), 
        class = "gar_SearchAnnotationSetsResponse")
}

#' Annotation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An annotation describes a region of reference genome. The value of an annotation may be one of several canonical types, supplemented by arbitrary info tags. An annotation is not inherently associated with a specific sample or individual (though a client could choose to use annotations in this way). Example canonical annotation types are `GENE` and `VARIANT`.
#' 
#' @param Annotation.info The \link{Annotation.info} object or list of objects
#' @param id The server-generated annotation ID, unique across all annotations
#' @param annotationSetId The annotation set to which this annotation belongs
#' @param name The display name of this annotation
#' @param referenceId The ID of the Google Genomics reference associated with this range
#' @param referenceName The display name corresponding to the reference specified by `referenceId`, for example `chr1`, `1`, or `chrX`
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param reverseStrand Whether this range refers to the reverse strand, as opposed to the forward strand
#' @param type The data type for this annotation
#' @param variant A variant annotation, which describes the effect of a variant on the genome, the coding sequence, and/or higher level consequences at the organism level e
#' @param transcript A transcript value represents the assertion that a particular region of the reference genome may be transcribed as RNA
#' @param info A map of additional read alignment information
#' 
#' @return Annotation object
#' 
#' @family Annotation functions
#' @export
Annotation <- function(Annotation.info = NULL, id = NULL, annotationSetId = NULL, 
    name = NULL, referenceId = NULL, referenceName = NULL, start = NULL, end = NULL, 
    reverseStrand = NULL, type = NULL, variant = NULL, transcript = NULL, info = NULL) {
    structure(list(Annotation.info = Annotation.info, id = id, annotationSetId = annotationSetId, 
        name = name, referenceId = referenceId, referenceName = referenceName, start = start, 
        end = end, reverseStrand = reverseStrand, type = type, variant = variant, 
        transcript = transcript, info = info), class = "gar_Annotation")
}

#' Annotation.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read alignment information. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return Annotation.info object
#' 
#' @family Annotation functions
#' @export
Annotation.info <- function() {
    list()
}

#' VariantAnnotation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param type Type has been adapted from ClinVar's list of variant types
#' @param effect Effect of the variant on the coding sequence
#' @param alternateBases The alternate allele for this variant
#' @param geneId Google annotation ID of the gene affected by this variant
#' @param transcriptIds Google annotation IDs of the transcripts affected by this variant
#' @param conditions The set of conditions associated with this variant
#' @param clinicalSignificance Describes the clinical significance of a variant
#' 
#' @return VariantAnnotation object
#' 
#' @family VariantAnnotation functions
#' @export
VariantAnnotation <- function(type = NULL, effect = NULL, alternateBases = NULL, 
    geneId = NULL, transcriptIds = NULL, conditions = NULL, clinicalSignificance = NULL) {
    structure(list(type = type, effect = effect, alternateBases = alternateBases, 
        geneId = geneId, transcriptIds = transcriptIds, conditions = conditions, 
        clinicalSignificance = clinicalSignificance), class = "gar_VariantAnnotation")
}

#' ClinicalCondition Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param names A set of names for the condition
#' @param externalIds The set of external IDs for this condition
#' @param conceptId The MedGen concept id associated with this gene
#' @param omimId The OMIM id for this condition
#' 
#' @return ClinicalCondition object
#' 
#' @family ClinicalCondition functions
#' @export
ClinicalCondition <- function(names = NULL, externalIds = NULL, conceptId = NULL, 
    omimId = NULL) {
    structure(list(names = names, externalIds = externalIds, conceptId = conceptId, 
        omimId = omimId), class = "gar_ClinicalCondition")
}

#' ExternalId Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param sourceName The name of the source of this data
#' @param id The id used by the source of this data
#' 
#' @return ExternalId object
#' 
#' @family ExternalId functions
#' @export
ExternalId <- function(sourceName = NULL, id = NULL) {
    structure(list(sourceName = sourceName, id = id), class = "gar_ExternalId")
}

#' Transcript Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A transcript represents the assertion that a particular region of the reference genome may be transcribed as RNA.
#' 
#' @param geneId The annotation ID of the gene from which this transcript is transcribed
#' @param exons The exons that compose this transcript
#' @param codingSequence The range of the coding sequence for this transcript, if any
#' 
#' @return Transcript object
#' 
#' @family Transcript functions
#' @export
Transcript <- function(geneId = NULL, exons = NULL, codingSequence = NULL) {
    structure(list(geneId = geneId, exons = exons, codingSequence = codingSequence), 
        class = "gar_Transcript")
}

#' Exon Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param start The start position of the exon on this annotation's reference sequence, 0-based inclusive
#' @param end The end position of the exon on this annotation's reference sequence, 0-based exclusive
#' @param frame The frame of this exon
#' 
#' @return Exon object
#' 
#' @family Exon functions
#' @export
Exon <- function(start = NULL, end = NULL, frame = NULL) {
    structure(list(start = start, end = end, frame = frame), class = "gar_Exon")
}

#' CodingSequence Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param start The start of the coding sequence on this annotation's reference sequence, 0-based inclusive
#' @param end The end of the coding sequence on this annotation's reference sequence, 0-based exclusive
#' 
#' @return CodingSequence object
#' 
#' @family CodingSequence functions
#' @export
CodingSequence <- function(start = NULL, end = NULL) {
    structure(list(start = start, end = end), class = "gar_CodingSequence")
}

#' BatchCreateAnnotationsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotations The annotations to be created
#' @param requestId A unique request ID which enables the server to detect duplicated requests
#' 
#' @return BatchCreateAnnotationsRequest object
#' 
#' @family BatchCreateAnnotationsRequest functions
#' @export
BatchCreateAnnotationsRequest <- function(annotations = NULL, requestId = NULL) {
    structure(list(annotations = annotations, requestId = requestId), class = "gar_BatchCreateAnnotationsRequest")
}

#' BatchCreateAnnotationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param entries The resulting per-annotation entries, ordered consistently with the original request
#' 
#' @return BatchCreateAnnotationsResponse object
#' 
#' @family BatchCreateAnnotationsResponse functions
#' @export
BatchCreateAnnotationsResponse <- function(entries = NULL) {
    structure(list(entries = entries), class = "gar_BatchCreateAnnotationsResponse")
}

#' Entry Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param status The creation status
#' @param annotation The created annotation, if creation was successful
#' 
#' @return Entry object
#' 
#' @family Entry functions
#' @export
Entry <- function(status = NULL, annotation = NULL) {
    structure(list(status = status, annotation = annotation), class = "gar_Entry")
}

#' Status Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The `Status` type defines a logical error model that is suitable for different programming environments, including REST APIs and RPC APIs. It is used by [gRPC](https://github.com/grpc). The error model is designed to be: - Simple to use and understand for most users - Flexible enough to meet unexpected needs # Overview The `Status` message contains three pieces of data: error code, error message, and error details. The error code should be an enum value of google.rpc.Code, but it may accept additional error codes if needed. The error message should be a developer-facing English message that helps developers *understand* and *resolve* the error. If a localized user-facing error message is needed, put the localized message in the error details or localize it in the client. The optional error details may contain arbitrary information about the error. There is a predefined set of error detail types in the package `google.rpc` which can be used for common error conditions. # Language mapping The `Status` message is the logical representation of the error model, but it is not necessarily the actual wire format. When the `Status` message is exposed in different client libraries and different wire protocols, it can be mapped differently. For example, it will likely be mapped to some exceptions in Java, but more likely mapped to some error codes in C. # Other uses The error model and the `Status` message can be used in a variety of environments, either with or without APIs, to provide a consistent developer experience across different environments. Example uses of this error model include: - Partial errors. If a service needs to return partial errors to the client, it may embed the `Status` in the normal response to indicate the partial errors. - Workflow errors. A typical workflow has multiple steps. Each step may have a `Status` message for error reporting purpose. - Batch operations. If a client uses batch request and batch response, the `Status` message should be used directly inside batch response, one for each error sub-response. - Asynchronous operations. If an API call embeds asynchronous operation results in its response, the status of those operations should be represented directly using the `Status` message. - Logging. If some API errors are stored in logs, the message `Status` could be used directly after any stripping needed for security/privacy reasons.
#' 
#' @param code The status code, which should be an enum value of google
#' @param message A developer-facing error message, which should be in English
#' @param details A list of messages that carry the error details
#' 
#' @return Status object
#' 
#' @family Status functions
#' @export
Status <- function(code = NULL, message = NULL, details = NULL) {
    structure(list(code = code, message = message, details = details), class = "gar_Status")
}

#' SearchAnnotationsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotationSetIds Required
#' @param referenceId The ID of the reference to query
#' @param referenceName The name of the reference to query, within the reference set associated with this query
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchAnnotationsRequest object
#' 
#' @family SearchAnnotationsRequest functions
#' @export
SearchAnnotationsRequest <- function(annotationSetIds = NULL, referenceId = NULL, 
    referenceName = NULL, start = NULL, end = NULL, pageToken = NULL, pageSize = NULL) {
    structure(list(annotationSetIds = annotationSetIds, referenceId = referenceId, 
        referenceName = referenceName, start = start, end = end, pageToken = pageToken, 
        pageSize = pageSize), class = "gar_SearchAnnotationsRequest")
}

#' SearchAnnotationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotations The matching annotations
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchAnnotationsResponse object
#' 
#' @family SearchAnnotationsResponse functions
#' @export
SearchAnnotationsResponse <- function(annotations = NULL, nextPageToken = NULL) {
    structure(list(annotations = annotations, nextPageToken = nextPageToken), class = "gar_SearchAnnotationsResponse")
}

#' ListDatasetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The dataset list response.
#' 
#' @param datasets The list of matching Datasets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return ListDatasetsResponse object
#' 
#' @family ListDatasetsResponse functions
#' @export
ListDatasetsResponse <- function(datasets = NULL, nextPageToken = NULL) {
    structure(list(datasets = datasets, nextPageToken = nextPageToken), class = "gar_ListDatasetsResponse")
}

#' Dataset Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A Dataset is a collection of genomic data. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param id The server-generated dataset ID, unique across all datasets
#' @param projectId The Google Developers Console project ID that this dataset belongs to
#' @param name The dataset name
#' @param createTime The time this dataset was created, in seconds from the epoch
#' 
#' @return Dataset object
#' 
#' @family Dataset functions
#' @export
Dataset <- function(id = NULL, projectId = NULL, name = NULL, createTime = NULL) {
    structure(list(id = id, projectId = projectId, name = name, createTime = createTime), 
        class = "gar_Dataset")
}

#' UndeleteDatasetRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' 
#' 
#' @return UndeleteDatasetRequest object
#' 
#' @family UndeleteDatasetRequest functions
#' @export
UndeleteDatasetRequest <- function() {
    list()
}

#' SetIamPolicyRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Request message for `SetIamPolicy` method.
#' 
#' @param policy REQUIRED: The complete policy to be applied to the `resource`
#' 
#' @return SetIamPolicyRequest object
#' 
#' @family SetIamPolicyRequest functions
#' @export
SetIamPolicyRequest <- function(policy = NULL) {
    structure(list(policy = policy), class = "gar_SetIamPolicyRequest")
}

#' Policy Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Defines an Identity and Access Management (IAM) policy. It is used to specify access control policies for Cloud Platform resources. A `Policy` consists of a list of `bindings`. A `Binding` binds a list of `members` to a `role`, where the members can be user accounts, Google groups, Google domains, and service accounts. A `role` is a named list of permissions defined by IAM. **Example** { 'bindings': [ { 'role': 'roles/owner', 'members': [ 'user:mike@example.com', 'group:admins@example.com', 'domain:google.com', 'serviceAccount:my-other-app@appspot.gserviceaccount.com', ] }, { 'role': 'roles/viewer', 'members': ['user:sean@example.com'] } ] } For a description of IAM and its features, see the [IAM developer's guide](https://cloud.google.com/iam).
#' 
#' @param version Version of the `Policy`
#' @param bindings Associates a list of `members` to a `role`
#' @param etag `etag` is used for optimistic concurrency control as a way to help prevent simultaneous updates of a policy from overwriting each other
#' 
#' @return Policy object
#' 
#' @family Policy functions
#' @export
Policy <- function(version = NULL, bindings = NULL, etag = NULL) {
    structure(list(version = version, bindings = bindings, etag = etag), class = "gar_Policy")
}

#' Binding Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Associates `members` with a `role`.
#' 
#' @param role Role that is assigned to `members`
#' @param members Specifies the identities requesting access for a Cloud Platform resource
#' 
#' @return Binding object
#' 
#' @family Binding functions
#' @export
Binding <- function(role = NULL, members = NULL) {
    structure(list(role = role, members = members), class = "gar_Binding")
}

#' GetIamPolicyRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Request message for `GetIamPolicy` method.
#' 
#' 
#' 
#' @return GetIamPolicyRequest object
#' 
#' @family GetIamPolicyRequest functions
#' @export
GetIamPolicyRequest <- function() {
    list()
}

#' TestIamPermissionsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Request message for `TestIamPermissions` method.
#' 
#' @param permissions REQUIRED: The set of permissions to check for the 'resource'
#' 
#' @return TestIamPermissionsRequest object
#' 
#' @family TestIamPermissionsRequest functions
#' @export
TestIamPermissionsRequest <- function(permissions = NULL) {
    structure(list(permissions = permissions), class = "gar_TestIamPermissionsRequest")
}

#' TestIamPermissionsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Response message for `TestIamPermissions` method.
#' 
#' @param permissions A subset of `TestPermissionsRequest
#' 
#' @return TestIamPermissionsResponse object
#' 
#' @family TestIamPermissionsResponse functions
#' @export
TestIamPermissionsResponse <- function(permissions = NULL) {
    structure(list(permissions = permissions), class = "gar_TestIamPermissionsResponse")
}

#' Operation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' This resource represents a long-running operation that is the result of a network API call.
#' 
#' @param Operation.metadata The \link{Operation.metadata} object or list of objects
#' @param Operation.response The \link{Operation.response} object or list of objects
#' @param name The server-assigned name, which is only unique within the same service that originally returns it
#' @param metadata An OperationMetadata object
#' @param done If the value is `false`, it means the operation is still in progress
#' @param error The error result of the operation in case of failure
#' @param response If importing ReadGroupSets, an ImportReadGroupSetsResponse is returned
#' 
#' @return Operation object
#' 
#' @family Operation functions
#' @export
Operation <- function(Operation.metadata = NULL, Operation.response = NULL, name = NULL, 
    metadata = NULL, done = NULL, error = NULL, response = NULL) {
    structure(list(Operation.metadata = Operation.metadata, Operation.response = Operation.response, 
        name = name, metadata = metadata, done = done, error = error, response = response), 
        class = "gar_Operation")
}

#' Operation.metadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An OperationMetadata object. This will always be returned with the Operation.
#' 
#' 
#' 
#' @return Operation.metadata object
#' 
#' @family Operation functions
#' @export
Operation.metadata <- function() {
    list()
}

#' Operation.response Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' If importing ReadGroupSets, an ImportReadGroupSetsResponse is returned. If importing Variants, an ImportVariantsResponse is returned. For exports, an empty response is returned.
#' 
#' 
#' 
#' @return Operation.response object
#' 
#' @family Operation functions
#' @export
Operation.response <- function() {
    list()
}

#' ListOperationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The response message for Operations.ListOperations.
#' 
#' @param operations A list of operations that matches the specified filter in the request
#' @param nextPageToken The standard List next-page token
#' 
#' @return ListOperationsResponse object
#' 
#' @family ListOperationsResponse functions
#' @export
ListOperationsResponse <- function(operations = NULL, nextPageToken = NULL) {
    structure(list(operations = operations, nextPageToken = nextPageToken), class = "gar_ListOperationsResponse")
}

#' CancelOperationRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The request message for Operations.CancelOperation.
#' 
#' 
#' 
#' @return CancelOperationRequest object
#' 
#' @family CancelOperationRequest functions
#' @export
CancelOperationRequest <- function() {
    list()
}

#' ImportReadGroupSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set import request.
#' 
#' @param datasetId Required
#' @param referenceSetId The reference set to which the imported read group sets are aligned to, if any
#' @param sourceUris A list of URIs pointing at [BAM files](https://samtools
#' @param partitionStrategy The partition strategy describes how read groups are partitioned into read group sets
#' 
#' @return ImportReadGroupSetsRequest object
#' 
#' @family ImportReadGroupSetsRequest functions
#' @export
ImportReadGroupSetsRequest <- function(datasetId = NULL, referenceSetId = NULL, sourceUris = NULL, 
    partitionStrategy = NULL) {
    structure(list(datasetId = datasetId, referenceSetId = referenceSetId, sourceUris = sourceUris, 
        partitionStrategy = partitionStrategy), class = "gar_ImportReadGroupSetsRequest")
}

#' ExportReadGroupSetRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set export request.
#' 
#' @param projectId Required
#' @param exportUri Required
#' @param referenceNames The reference names to export
#' 
#' @return ExportReadGroupSetRequest object
#' 
#' @family ExportReadGroupSetRequest functions
#' @export
ExportReadGroupSetRequest <- function(projectId = NULL, exportUri = NULL, referenceNames = NULL) {
    structure(list(projectId = projectId, exportUri = exportUri, referenceNames = referenceNames), 
        class = "gar_ExportReadGroupSetRequest")
}

#' SearchReadGroupSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set search request.
#' 
#' @param datasetIds Restricts this query to read group sets within the given datasets
#' @param name Only return read group sets for which a substring of the name matches this string
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReadGroupSetsRequest object
#' 
#' @family SearchReadGroupSetsRequest functions
#' @export
SearchReadGroupSetsRequest <- function(datasetIds = NULL, name = NULL, pageToken = NULL, 
    pageSize = NULL) {
    structure(list(datasetIds = datasetIds, name = name, pageToken = pageToken, pageSize = pageSize), 
        class = "gar_SearchReadGroupSetsRequest")
}

#' SearchReadGroupSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set search response.
#' 
#' @param readGroupSets The list of matching read group sets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReadGroupSetsResponse object
#' 
#' @family SearchReadGroupSetsResponse functions
#' @export
SearchReadGroupSetsResponse <- function(readGroupSets = NULL, nextPageToken = NULL) {
    structure(list(readGroupSets = readGroupSets, nextPageToken = nextPageToken), 
        class = "gar_SearchReadGroupSetsResponse")
}

#' ReadGroupSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read group set is a logical collection of read groups, which are collections of reads produced by a sequencer. A read group set typically models reads corresponding to one sample, sequenced one way, and aligned one way. * A read group set belongs to one dataset. * A read group belongs to one read group set. * A read belongs to one read group. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param ReadGroupSet.info The \link{ReadGroupSet.info} object or list of objects
#' @param id The server-generated read group set ID, unique for all read group sets
#' @param datasetId The dataset to which this read group set belongs
#' @param referenceSetId The reference set to which the reads in this read group set are aligned
#' @param name The read group set name
#' @param filename The filename of the original source file for this read group set, if any
#' @param readGroups The read groups in this set
#' @param info A map of additional read group set information
#' 
#' @return ReadGroupSet object
#' 
#' @family ReadGroupSet functions
#' @export
ReadGroupSet <- function(ReadGroupSet.info = NULL, id = NULL, datasetId = NULL, referenceSetId = NULL, 
    name = NULL, filename = NULL, readGroups = NULL, info = NULL) {
    structure(list(ReadGroupSet.info = ReadGroupSet.info, id = id, datasetId = datasetId, 
        referenceSetId = referenceSetId, name = name, filename = filename, readGroups = readGroups, 
        info = info), class = "gar_ReadGroupSet")
}

#' ReadGroupSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read group set information.
#' 
#' 
#' 
#' @return ReadGroupSet.info object
#' 
#' @family ReadGroupSet functions
#' @export
ReadGroupSet.info <- function() {
    list()
}

#' ReadGroup Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read group is all the data that's processed the same way by the sequencer.
#' 
#' @param ReadGroup.info The \link{ReadGroup.info} object or list of objects
#' @param id The server-generated read group ID, unique for all read groups
#' @param datasetId The dataset to which this read group belongs
#' @param name The read group name
#' @param description A free-form text description of this read group
#' @param sampleId A client-supplied sample identifier for the reads in this read group
#' @param experiment The experiment used to generate this read group
#' @param predictedInsertSize The predicted insert size of this read group
#' @param programs The programs used to generate this read group
#' @param referenceSetId The reference set the reads in this read group are aligned to
#' @param info A map of additional read group information
#' 
#' @return ReadGroup object
#' 
#' @family ReadGroup functions
#' @export
ReadGroup <- function(ReadGroup.info = NULL, id = NULL, datasetId = NULL, name = NULL, 
    description = NULL, sampleId = NULL, experiment = NULL, predictedInsertSize = NULL, 
    programs = NULL, referenceSetId = NULL, info = NULL) {
    structure(list(ReadGroup.info = ReadGroup.info, id = id, datasetId = datasetId, 
        name = name, description = description, sampleId = sampleId, experiment = experiment, 
        predictedInsertSize = predictedInsertSize, programs = programs, referenceSetId = referenceSetId, 
        info = info), class = "gar_ReadGroup")
}

#' ReadGroup.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read group information. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return ReadGroup.info object
#' 
#' @family ReadGroup functions
#' @export
ReadGroup.info <- function() {
    list()
}

#' Experiment Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param libraryId A client-supplied library identifier; a library is a collection of DNA fragments which have been prepared for sequencing from a sample
#' @param platformUnit The platform unit used as part of this experiment, for example flowcell-barcode
#' @param sequencingCenter The sequencing center used as part of this experiment
#' @param instrumentModel The instrument model used as part of this experiment
#' 
#' @return Experiment object
#' 
#' @family Experiment functions
#' @export
Experiment <- function(libraryId = NULL, platformUnit = NULL, sequencingCenter = NULL, 
    instrumentModel = NULL) {
    structure(list(libraryId = libraryId, platformUnit = platformUnit, sequencingCenter = sequencingCenter, 
        instrumentModel = instrumentModel), class = "gar_Experiment")
}

#' Program Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param commandLine The command line used to run this program
#' @param id The user specified locally unique ID of the program
#' @param name The display name of the program
#' @param prevProgramId The ID of the program run before this one
#' @param version The version of the program run
#' 
#' @return Program object
#' 
#' @family Program functions
#' @export
Program <- function(commandLine = NULL, id = NULL, name = NULL, prevProgramId = NULL, 
    version = NULL) {
    structure(list(commandLine = commandLine, id = id, name = name, prevProgramId = prevProgramId, 
        version = version), class = "gar_Program")
}

#' ListCoverageBucketsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param bucketWidth The length of each coverage bucket in base pairs
#' @param coverageBuckets The coverage buckets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return ListCoverageBucketsResponse object
#' 
#' @family ListCoverageBucketsResponse functions
#' @export
ListCoverageBucketsResponse <- function(bucketWidth = NULL, coverageBuckets = NULL, 
    nextPageToken = NULL) {
    structure(list(bucketWidth = bucketWidth, coverageBuckets = coverageBuckets, 
        nextPageToken = nextPageToken), class = "gar_ListCoverageBucketsResponse")
}

#' CoverageBucket Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A bucket over which read coverage has been precomputed. A bucket corresponds to a specific range of the reference sequence.
#' 
#' @param range The genomic coordinate range spanned by this bucket
#' @param meanCoverage The average number of reads which are aligned to each individual reference base in this bucket
#' 
#' @return CoverageBucket object
#' 
#' @family CoverageBucket functions
#' @export
CoverageBucket <- function(range = NULL, meanCoverage = NULL) {
    structure(list(range = range, meanCoverage = meanCoverage), class = "gar_CoverageBucket")
}

#' Range Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A 0-based half-open genomic coordinate range for search requests.
#' 
#' @param referenceName The reference sequence name, for example `chr1`, `1`, or `chrX`
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param end The end position of the range on the reference, 0-based exclusive
#' 
#' @return Range object
#' 
#' @family Range functions
#' @export
Range <- function(referenceName = NULL, start = NULL, end = NULL) {
    structure(list(referenceName = referenceName, start = start, end = end), class = "gar_Range")
}

#' SearchReadsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read search request.
#' 
#' @param readGroupSetIds The IDs of the read groups sets within which to search for reads
#' @param readGroupIds The IDs of the read groups within which to search for reads
#' @param referenceName The reference sequence name, for example `chr1`, `1`, or `chrX`
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReadsRequest object
#' 
#' @family SearchReadsRequest functions
#' @export
SearchReadsRequest <- function(readGroupSetIds = NULL, readGroupIds = NULL, referenceName = NULL, 
    start = NULL, end = NULL, pageToken = NULL, pageSize = NULL) {
    structure(list(readGroupSetIds = readGroupSetIds, readGroupIds = readGroupIds, 
        referenceName = referenceName, start = start, end = end, pageToken = pageToken, 
        pageSize = pageSize), class = "gar_SearchReadsRequest")
}

#' SearchReadsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read search response.
#' 
#' @param alignments The list of matching alignments sorted by mapped genomic coordinate, if any, ascending in position within the same reference
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReadsResponse object
#' 
#' @family SearchReadsResponse functions
#' @export
SearchReadsResponse <- function(alignments = NULL, nextPageToken = NULL) {
    structure(list(alignments = alignments, nextPageToken = nextPageToken), class = "gar_SearchReadsResponse")
}

#' Read Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read alignment describes a linear alignment of a string of DNA to a reference sequence, in addition to metadata about the fragment (the molecule of DNA sequenced) and the read (the bases which were read by the sequencer). A read is equivalent to a line in a SAM file. A read belongs to exactly one read group and exactly one read group set. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) ### Reverse-stranded reads Mapped reads (reads having a non-null `alignment`) can be aligned to either the forward or the reverse strand of their associated reference. Strandedness of a mapped read is encoded by `alignment.position.reverseStrand`. If we consider the reference to be a forward-stranded coordinate space of `[0, reference.length)` with `0` as the left-most position and `reference.length` as the right-most position, reads are always aligned left to right. That is, `alignment.position.position` always refers to the left-most reference coordinate and `alignment.cigar` describes the alignment of this read to the reference from left to right. All per-base fields such as `alignedSequence` and `alignedQuality` share this same left-to-right orientation; this is true of reads which are aligned to either strand. For reverse-stranded reads, this means that `alignedSequence` is the reverse complement of the bases that were originally reported by the sequencing machine. ### Generating a reference-aligned sequence string When interacting with mapped reads, it's often useful to produce a string representing the local alignment of the read to reference. The following pseudocode demonstrates one way of doing this: out = '' offset = 0 for c in read.alignment.cigar { switch c.operation { case 'ALIGNMENT_MATCH', 'SEQUENCE_MATCH', 'SEQUENCE_MISMATCH': out += read.alignedSequence[offset:offset+c.operationLength] offset += c.operationLength break case 'CLIP_SOFT', 'INSERT': offset += c.operationLength break case 'PAD': out += repeat('*', c.operationLength) break case 'DELETE': out += repeat('-', c.operationLength) break case 'SKIP': out += repeat(' ', c.operationLength) break case 'CLIP_HARD': break } } return out ### Converting to SAM's CIGAR string The following pseudocode generates a SAM CIGAR string from the `cigar` field. Note that this is a lossy conversion (`cigar.referenceSequence` is lost). cigarMap = { 'ALIGNMENT_MATCH': 'M', 'INSERT': 'I', 'DELETE': 'D', 'SKIP': 'N', 'CLIP_SOFT': 'S', 'CLIP_HARD': 'H', 'PAD': 'P', 'SEQUENCE_MATCH': '=', 'SEQUENCE_MISMATCH': 'X', } cigarStr = '' for c in read.alignment.cigar { cigarStr += c.operationLength + cigarMap[c.operation] } return cigarStr
#' 
#' @param Read.info The \link{Read.info} object or list of objects
#' @param id The server-generated read ID, unique across all reads
#' @param readGroupId The ID of the read group this read belongs to
#' @param readGroupSetId The ID of the read group set this read belongs to
#' @param fragmentName The fragment name
#' @param properPlacement The orientation and the distance between reads from the fragment are consistent with the sequencing protocol (SAM flag 0x2)
#' @param duplicateFragment The fragment is a PCR or optical duplicate (SAM flag 0x400)
#' @param fragmentLength The observed length of the fragment, equivalent to TLEN in SAM
#' @param readNumber The read number in sequencing
#' @param numberReads The number of reads in the fragment (extension to SAM flag 0x1)
#' @param failedVendorQualityChecks Whether this read did not pass filters, such as platform or vendor quality controls (SAM flag 0x200)
#' @param alignment The linear alignment for this alignment record
#' @param secondaryAlignment Whether this alignment is secondary
#' @param supplementaryAlignment Whether this alignment is supplementary
#' @param alignedSequence The bases of the read sequence contained in this alignment record, **without CIGAR operations applied** (equivalent to SEQ in SAM)
#' @param alignedQuality The quality of the read sequence contained in this alignment record (equivalent to QUAL in SAM)
#' @param nextMatePosition The mapping of the primary alignment of the `(readNumber+1)%numberReads` read in the fragment
#' @param info A map of additional read alignment information
#' 
#' @return Read object
#' 
#' @family Read functions
#' @export
Read <- function(Read.info = NULL, id = NULL, readGroupId = NULL, readGroupSetId = NULL, 
    fragmentName = NULL, properPlacement = NULL, duplicateFragment = NULL, fragmentLength = NULL, 
    readNumber = NULL, numberReads = NULL, failedVendorQualityChecks = NULL, alignment = NULL, 
    secondaryAlignment = NULL, supplementaryAlignment = NULL, alignedSequence = NULL, 
    alignedQuality = NULL, nextMatePosition = NULL, info = NULL) {
    structure(list(Read.info = Read.info, id = id, readGroupId = readGroupId, readGroupSetId = readGroupSetId, 
        fragmentName = fragmentName, properPlacement = properPlacement, duplicateFragment = duplicateFragment, 
        fragmentLength = fragmentLength, readNumber = readNumber, numberReads = numberReads, 
        failedVendorQualityChecks = failedVendorQualityChecks, alignment = alignment, 
        secondaryAlignment = secondaryAlignment, supplementaryAlignment = supplementaryAlignment, 
        alignedSequence = alignedSequence, alignedQuality = alignedQuality, nextMatePosition = nextMatePosition, 
        info = info), class = "gar_Read")
}

#' Read.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional read alignment information. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return Read.info object
#' 
#' @family Read functions
#' @export
Read.info <- function() {
    list()
}

#' LinearAlignment Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A linear alignment can be represented by one CIGAR string. Describes the mapped position and local alignment of the read to the reference.
#' 
#' @param position The position of this alignment
#' @param mappingQuality The mapping quality of this alignment
#' @param cigar Represents the local alignment of this sequence (alignment matches, indels, etc) against the reference
#' 
#' @return LinearAlignment object
#' 
#' @family LinearAlignment functions
#' @export
LinearAlignment <- function(position = NULL, mappingQuality = NULL, cigar = NULL) {
    structure(list(position = position, mappingQuality = mappingQuality, cigar = cigar), 
        class = "gar_LinearAlignment")
}

#' Position Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An abstraction for referring to a genomic position, in relation to some already known reference. For now, represents a genomic position as a reference name, a base number on that reference (0-based), and a determination of forward or reverse strand.
#' 
#' @param referenceName The name of the reference in whatever reference set is being used
#' @param position The 0-based offset from the start of the forward strand for that reference
#' @param reverseStrand Whether this position is on the reverse strand, as opposed to the forward strand
#' 
#' @return Position object
#' 
#' @family Position functions
#' @export
Position <- function(referenceName = NULL, position = NULL, reverseStrand = NULL) {
    structure(list(referenceName = referenceName, position = position, reverseStrand = reverseStrand), 
        class = "gar_Position")
}

#' CigarUnit Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A single CIGAR operation.
#' 
#' @param operation No description
#' @param operationLength The number of genomic bases that the operation runs for
#' @param referenceSequence `referenceSequence` is only used at mismatches (`SEQUENCE_MISMATCH`) and deletions (`DELETE`)
#' 
#' @return CigarUnit object
#' 
#' @family CigarUnit functions
#' @export
CigarUnit <- function(operation = NULL, operationLength = NULL, referenceSequence = NULL) {
    structure(list(operation = operation, operationLength = operationLength, referenceSequence = referenceSequence), 
        class = "gar_CigarUnit")
}

#' StreamReadsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The stream reads request.
#' 
#' @param projectId The Google Developers Console project ID or number which will be billed for this access
#' @param readGroupSetId The ID of the read group set from which to stream reads
#' @param referenceName The reference sequence name, for example `chr1`, `1`, or `chrX`
#' @param start The start position of the range on the reference, 0-based inclusive
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param shard Restricts results to a shard containing approximately `1/totalShards` of the normal response payload for this query
#' @param totalShards Specifying `totalShards` causes a disjoint subset of the normal response payload to be returned for each query with a unique `shard` parameter specified
#' 
#' @return StreamReadsRequest object
#' 
#' @family StreamReadsRequest functions
#' @export
StreamReadsRequest <- function(projectId = NULL, readGroupSetId = NULL, referenceName = NULL, 
    start = NULL, end = NULL, shard = NULL, totalShards = NULL) {
    structure(list(projectId = projectId, readGroupSetId = readGroupSetId, referenceName = referenceName, 
        start = start, end = end, shard = shard, totalShards = totalShards), class = "gar_StreamReadsRequest")
}

#' StreamReadsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param alignments No description
#' 
#' @return StreamReadsResponse object
#' 
#' @family StreamReadsResponse functions
#' @export
StreamReadsResponse <- function(alignments = NULL) {
    structure(list(alignments = alignments), class = "gar_StreamReadsResponse")
}

#' SearchReferenceSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param md5checksums If present, return reference sets for which the md5checksum matches exactly
#' @param accessions If present, return reference sets for which a prefix of any of sourceAccessions match any of these strings
#' @param assemblyId If present, return reference sets for which a substring of their `assemblyId` matches this string (case insensitive)
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReferenceSetsRequest object
#' 
#' @family SearchReferenceSetsRequest functions
#' @export
SearchReferenceSetsRequest <- function(md5checksums = NULL, accessions = NULL, assemblyId = NULL, 
    pageToken = NULL, pageSize = NULL) {
    structure(list(md5checksums = md5checksums, accessions = accessions, assemblyId = assemblyId, 
        pageToken = pageToken, pageSize = pageSize), class = "gar_SearchReferenceSetsRequest")
}

#' SearchReferenceSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param referenceSets The matching references sets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReferenceSetsResponse object
#' 
#' @family SearchReferenceSetsResponse functions
#' @export
SearchReferenceSetsResponse <- function(referenceSets = NULL, nextPageToken = NULL) {
    structure(list(referenceSets = referenceSets, nextPageToken = nextPageToken), 
        class = "gar_SearchReferenceSetsResponse")
}

#' ReferenceSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A reference set is a set of references which typically comprise a reference assembly for a species, such as `GRCh38` which is representative of the human genome. A reference set defines a common coordinate space for comparing reference-aligned experimental data. A reference set contains 1 or more references. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param id The server-generated reference set ID, unique across all reference sets
#' @param referenceIds The IDs of the reference objects that are part of this set
#' @param md5checksum Order-independent MD5 checksum which identifies this reference set
#' @param ncbiTaxonId ID from http://www
#' @param description Free text description of this reference set
#' @param assemblyId Public id of this reference set, such as `GRCh37`
#' @param sourceUri The URI from which the references were obtained
#' @param sourceAccessions All known corresponding accession IDs in INSDC (GenBank/ENA/DDBJ) ideally with a version number, for example `NC_000001
#' 
#' @return ReferenceSet object
#' 
#' @family ReferenceSet functions
#' @export
ReferenceSet <- function(id = NULL, referenceIds = NULL, md5checksum = NULL, ncbiTaxonId = NULL, 
    description = NULL, assemblyId = NULL, sourceUri = NULL, sourceAccessions = NULL) {
    structure(list(id = id, referenceIds = referenceIds, md5checksum = md5checksum, 
        ncbiTaxonId = ncbiTaxonId, description = description, assemblyId = assemblyId, 
        sourceUri = sourceUri, sourceAccessions = sourceAccessions), class = "gar_ReferenceSet")
}

#' SearchReferencesRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param md5checksums If present, return references for which the md5checksum matches exactly
#' @param accessions If present, return references for which a prefix of any of sourceAccessions match any of these strings
#' @param referenceSetId If present, return only references which belong to this reference set
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchReferencesRequest object
#' 
#' @family SearchReferencesRequest functions
#' @export
SearchReferencesRequest <- function(md5checksums = NULL, accessions = NULL, referenceSetId = NULL, 
    pageToken = NULL, pageSize = NULL) {
    structure(list(md5checksums = md5checksums, accessions = accessions, referenceSetId = referenceSetId, 
        pageToken = pageToken, pageSize = pageSize), class = "gar_SearchReferencesRequest")
}

#' SearchReferencesResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param references The matching references
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReferencesResponse object
#' 
#' @family SearchReferencesResponse functions
#' @export
SearchReferencesResponse <- function(references = NULL, nextPageToken = NULL) {
    structure(list(references = references, nextPageToken = nextPageToken), class = "gar_SearchReferencesResponse")
}

#' Reference Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A reference is a canonical assembled DNA sequence, intended to act as a reference coordinate space for other genomic annotations. A single reference might represent the human chromosome 1 or mitochandrial DNA, for instance. A reference belongs to one or more reference sets. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param id The server-generated reference ID, unique across all references
#' @param length The length of this reference's sequence
#' @param md5checksum MD5 of the upper-case sequence excluding all whitespace characters (this is equivalent to SQ:M5 in SAM)
#' @param name The name of this reference, for example `22`
#' @param sourceUri The URI from which the sequence was obtained
#' @param sourceAccessions All known corresponding accession IDs in INSDC (GenBank/ENA/DDBJ) ideally with a version number, for example `GCF_000001405
#' @param ncbiTaxonId ID from http://www
#' 
#' @return Reference object
#' 
#' @family Reference functions
#' @export
Reference <- function(id = NULL, length = NULL, md5checksum = NULL, name = NULL, 
    sourceUri = NULL, sourceAccessions = NULL, ncbiTaxonId = NULL) {
    structure(list(id = id, length = length, md5checksum = md5checksum, name = name, 
        sourceUri = sourceUri, sourceAccessions = sourceAccessions, ncbiTaxonId = ncbiTaxonId), 
        class = "gar_Reference")
}

#' ListBasesResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param offset The offset position (0-based) of the given `sequence` from the start of this `Reference`
#' @param sequence A substring of the bases that make up this reference
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return ListBasesResponse object
#' 
#' @family ListBasesResponse functions
#' @export
ListBasesResponse <- function(offset = NULL, sequence = NULL, nextPageToken = NULL) {
    structure(list(offset = offset, sequence = sequence, nextPageToken = nextPageToken), 
        class = "gar_ListBasesResponse")
}

#' ImportVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data import request.
#' 
#' @param ImportVariantsRequest.infoMergeConfig The \link{ImportVariantsRequest.infoMergeConfig} object or list of objects
#' @param variantSetId Required
#' @param sourceUris A list of URIs referencing variant files in Google Cloud Storage
#' @param format The format of the variant data being imported
#' @param normalizeReferenceNames Convert reference names to the canonical representation
#' @param infoMergeConfig A mapping between info field keys and the InfoMergeOperations to be performed on them
#' 
#' @return ImportVariantsRequest object
#' 
#' @family ImportVariantsRequest functions
#' @export
ImportVariantsRequest <- function(ImportVariantsRequest.infoMergeConfig = NULL, variantSetId = NULL, 
    sourceUris = NULL, format = NULL, normalizeReferenceNames = NULL, infoMergeConfig = NULL) {
    structure(list(ImportVariantsRequest.infoMergeConfig = ImportVariantsRequest.infoMergeConfig, 
        variantSetId = variantSetId, sourceUris = sourceUris, format = format, normalizeReferenceNames = normalizeReferenceNames, 
        infoMergeConfig = infoMergeConfig), class = "gar_ImportVariantsRequest")
}

#' ImportVariantsRequest.infoMergeConfig Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A mapping between info field keys and the InfoMergeOperations to be performed on them. This is plumbed down to the MergeVariantRequests generated by the resulting import job.
#' 
#' 
#' 
#' @return ImportVariantsRequest.infoMergeConfig object
#' 
#' @family ImportVariantsRequest functions
#' @export
ImportVariantsRequest.infoMergeConfig <- function() {
    list()
}

#' VariantSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A variant set is a collection of call sets and variants. It contains summary statistics of those contents. A variant set belongs to a dataset. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param datasetId The dataset to which this variant set belongs
#' @param id The server-generated variant set ID, unique across all variant sets
#' @param referenceSetId The reference set to which the variant set is mapped
#' @param referenceBounds A list of all references used by the variants in a variant set with associated coordinate upper bounds for each one
#' @param metadata The metadata associated with this variant set
#' @param name User-specified, mutable name
#' @param description A textual description of this variant set
#' 
#' @return VariantSet object
#' 
#' @family VariantSet functions
#' @export
VariantSet <- function(datasetId = NULL, id = NULL, referenceSetId = NULL, referenceBounds = NULL, 
    metadata = NULL, name = NULL, description = NULL) {
    structure(list(datasetId = datasetId, id = id, referenceSetId = referenceSetId, 
        referenceBounds = referenceBounds, metadata = metadata, name = name, description = description), 
        class = "gar_VariantSet")
}

#' ReferenceBound Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' ReferenceBound records an upper bound for the starting coordinate of variants in a particular reference.
#' 
#' @param referenceName The name of the reference associated with this reference bound
#' @param upperBound An upper bound (inclusive) on the starting coordinate of any variant in the reference sequence
#' 
#' @return ReferenceBound object
#' 
#' @family ReferenceBound functions
#' @export
ReferenceBound <- function(referenceName = NULL, upperBound = NULL) {
    structure(list(referenceName = referenceName, upperBound = upperBound), class = "gar_ReferenceBound")
}

#' VariantSetMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Metadata describes a single piece of variant call metadata. These data include a top level key and either a single value string (value) or a list of key-value pairs (info.) Value and info are mutually exclusive.
#' 
#' @param VariantSetMetadata.info The \link{VariantSetMetadata.info} object or list of objects
#' @param key The top-level key
#' @param value The value field for simple metadata
#' @param id User-provided ID field, not enforced by this API
#' @param type The type of data
#' @param number The number of values that can be included in a field described by this metadata
#' @param description A textual description of this metadata
#' @param info Remaining structured metadata key-value pairs
#' 
#' @return VariantSetMetadata object
#' 
#' @family VariantSetMetadata functions
#' @export
VariantSetMetadata <- function(VariantSetMetadata.info = NULL, key = NULL, value = NULL, 
    id = NULL, type = NULL, number = NULL, description = NULL, info = NULL) {
    structure(list(VariantSetMetadata.info = VariantSetMetadata.info, key = key, 
        value = value, id = id, type = type, number = number, description = description, 
        info = info), class = "gar_VariantSetMetadata")
}

#' VariantSetMetadata.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Remaining structured metadata key-value pairs. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return VariantSetMetadata.info object
#' 
#' @family VariantSetMetadata functions
#' @export
VariantSetMetadata.info <- function() {
    list()
}

#' ExportVariantSetRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data export request.
#' 
#' @param callSetIds If provided, only variant call information from the specified call sets will be exported
#' @param projectId Required
#' @param format The format for the exported data
#' @param bigqueryDataset Required
#' @param bigqueryTable Required
#' 
#' @return ExportVariantSetRequest object
#' 
#' @family ExportVariantSetRequest functions
#' @export
ExportVariantSetRequest <- function(callSetIds = NULL, projectId = NULL, format = NULL, 
    bigqueryDataset = NULL, bigqueryTable = NULL) {
    structure(list(callSetIds = callSetIds, projectId = projectId, format = format, 
        bigqueryDataset = bigqueryDataset, bigqueryTable = bigqueryTable), class = "gar_ExportVariantSetRequest")
}

#' SearchVariantSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The search variant sets request.
#' 
#' @param datasetIds Exactly one dataset ID must be provided here
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchVariantSetsRequest object
#' 
#' @family SearchVariantSetsRequest functions
#' @export
SearchVariantSetsRequest <- function(datasetIds = NULL, pageToken = NULL, pageSize = NULL) {
    structure(list(datasetIds = datasetIds, pageToken = pageToken, pageSize = pageSize), 
        class = "gar_SearchVariantSetsRequest")
}

#' SearchVariantSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The search variant sets response.
#' 
#' @param variantSets The variant sets belonging to the requested dataset
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchVariantSetsResponse object
#' 
#' @family SearchVariantSetsResponse functions
#' @export
SearchVariantSetsResponse <- function(variantSets = NULL, nextPageToken = NULL) {
    structure(list(variantSets = variantSets, nextPageToken = nextPageToken), class = "gar_SearchVariantSetsResponse")
}

#' SearchVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant search request.
#' 
#' @param variantSetIds At most one variant set ID must be provided
#' @param variantName Only return variants which have exactly this name
#' @param callSetIds Only return variant calls which belong to call sets with these ids
#' @param referenceName Required
#' @param start The beginning of the window (0-based, inclusive) for which overlapping variants should be returned
#' @param end The end of the window, 0-based exclusive
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of variants to return in a single page
#' @param maxCalls The maximum number of calls to return in a single page
#' 
#' @return SearchVariantsRequest object
#' 
#' @family SearchVariantsRequest functions
#' @export
SearchVariantsRequest <- function(variantSetIds = NULL, variantName = NULL, callSetIds = NULL, 
    referenceName = NULL, start = NULL, end = NULL, pageToken = NULL, pageSize = NULL, 
    maxCalls = NULL) {
    structure(list(variantSetIds = variantSetIds, variantName = variantName, callSetIds = callSetIds, 
        referenceName = referenceName, start = start, end = end, pageToken = pageToken, 
        pageSize = pageSize, maxCalls = maxCalls), class = "gar_SearchVariantsRequest")
}

#' SearchVariantsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant search response.
#' 
#' @param variants The list of matching Variants
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchVariantsResponse object
#' 
#' @family SearchVariantsResponse functions
#' @export
SearchVariantsResponse <- function(variants = NULL, nextPageToken = NULL) {
    structure(list(variants = variants, nextPageToken = nextPageToken), class = "gar_SearchVariantsResponse")
}

#' Variant Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A variant represents a change in DNA sequence relative to a reference sequence. For example, a variant could represent a SNP or an insertion. Variants belong to a variant set. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics) Each of the calls on a variant represent a determination of genotype with respect to that variant. For example, a call might assign probability of 0.32 to the occurrence of a SNP named rs1234 in a sample named NA12345. A call belongs to a call set, which contains related calls typically from one sample.
#' 
#' @param Variant.info The \link{Variant.info} object or list of objects
#' @param variantSetId The ID of the variant set this variant belongs to
#' @param id The server-generated variant ID, unique across all variants
#' @param names Names for the variant, for example a RefSNP ID
#' @param created The date this variant was created, in milliseconds from the epoch
#' @param referenceName The reference on which this variant occurs
#' @param start The position at which this variant occurs (0-based)
#' @param end The end position (0-based) of this variant
#' @param referenceBases The reference bases for this variant
#' @param alternateBases The bases that appear instead of the reference bases
#' @param quality A measure of how likely this variant is to be real
#' @param filter A list of filters (normally quality filters) this variant has failed
#' @param info A map of additional variant information
#' @param calls The variant calls for this particular variant
#' 
#' @return Variant object
#' 
#' @family Variant functions
#' @export
Variant <- function(Variant.info = NULL, variantSetId = NULL, id = NULL, names = NULL, 
    created = NULL, referenceName = NULL, start = NULL, end = NULL, referenceBases = NULL, 
    alternateBases = NULL, quality = NULL, filter = NULL, info = NULL, calls = NULL) {
    structure(list(Variant.info = Variant.info, variantSetId = variantSetId, id = id, 
        names = names, created = created, referenceName = referenceName, start = start, 
        end = end, referenceBases = referenceBases, alternateBases = alternateBases, 
        quality = quality, filter = filter, info = info, calls = calls), class = "gar_Variant")
}

#' Variant.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional variant information. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return Variant.info object
#' 
#' @family Variant functions
#' @export
Variant.info <- function() {
    list()
}

#' VariantCall Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a call set with the name NA12345.
#' 
#' @param VariantCall.info The \link{VariantCall.info} object or list of objects
#' @param callSetId The ID of the call set this variant call belongs to
#' @param callSetName The name of the call set this variant call belongs to
#' @param genotype The genotype of this variant call
#' @param phaseset If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls in the same reference sequence which have the same phaseset value
#' @param genotypeLikelihood The genotype likelihoods for this variant call
#' @param info A map of additional variant call information
#' 
#' @return VariantCall object
#' 
#' @family VariantCall functions
#' @export
VariantCall <- function(VariantCall.info = NULL, callSetId = NULL, callSetName = NULL, 
    genotype = NULL, phaseset = NULL, genotypeLikelihood = NULL, info = NULL) {
    structure(list(VariantCall.info = VariantCall.info, callSetId = callSetId, callSetName = callSetName, 
        genotype = genotype, phaseset = phaseset, genotypeLikelihood = genotypeLikelihood, 
        info = info), class = "gar_VariantCall")
}

#' VariantCall.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional variant call information. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return VariantCall.info object
#' 
#' @family VariantCall functions
#' @export
VariantCall.info <- function() {
    list()
}

#' MergeVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param MergeVariantsRequest.infoMergeConfig The \link{MergeVariantsRequest.infoMergeConfig} object or list of objects
#' @param variantSetId The destination variant set
#' @param variants The variants to be merged with existing variants
#' @param infoMergeConfig A mapping between info field keys and the InfoMergeOperations to be performed on them
#' 
#' @return MergeVariantsRequest object
#' 
#' @family MergeVariantsRequest functions
#' @export
MergeVariantsRequest <- function(MergeVariantsRequest.infoMergeConfig = NULL, variantSetId = NULL, 
    variants = NULL, infoMergeConfig = NULL) {
    structure(list(MergeVariantsRequest.infoMergeConfig = MergeVariantsRequest.infoMergeConfig, 
        variantSetId = variantSetId, variants = variants, infoMergeConfig = infoMergeConfig), 
        class = "gar_MergeVariantsRequest")
}

#' MergeVariantsRequest.infoMergeConfig Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A mapping between info field keys and the InfoMergeOperations to be performed on them.
#' 
#' 
#' 
#' @return MergeVariantsRequest.infoMergeConfig object
#' 
#' @family MergeVariantsRequest functions
#' @export
MergeVariantsRequest.infoMergeConfig <- function() {
    list()
}

#' SearchCallSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The call set search request.
#' 
#' @param variantSetIds Restrict the query to call sets within the given variant sets
#' @param name Only return call sets for which a substring of the name matches this string
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param pageSize The maximum number of results to return in a single page
#' 
#' @return SearchCallSetsRequest object
#' 
#' @family SearchCallSetsRequest functions
#' @export
SearchCallSetsRequest <- function(variantSetIds = NULL, name = NULL, pageToken = NULL, 
    pageSize = NULL) {
    structure(list(variantSetIds = variantSetIds, name = name, pageToken = pageToken, 
        pageSize = pageSize), class = "gar_SearchCallSetsRequest")
}

#' SearchCallSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The call set search response.
#' 
#' @param callSets The list of matching call sets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchCallSetsResponse object
#' 
#' @family SearchCallSetsResponse functions
#' @export
SearchCallSetsResponse <- function(callSets = NULL, nextPageToken = NULL) {
    structure(list(callSets = callSets, nextPageToken = nextPageToken), class = "gar_SearchCallSetsResponse")
}

#' CallSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A call set is a collection of variant calls, typically for one sample. It belongs to a variant set. For more genomics resource definitions, see [Fundamentals of Google Genomics](https://cloud.google.com/genomics/fundamentals-of-google-genomics)
#' 
#' @param CallSet.info The \link{CallSet.info} object or list of objects
#' @param id The server-generated call set ID, unique across all call sets
#' @param name The call set name
#' @param sampleId The sample ID this call set corresponds to
#' @param variantSetIds The IDs of the variant sets this call set belongs to
#' @param created The date this call set was created in milliseconds from the epoch
#' @param info A map of additional call set information
#' 
#' @return CallSet object
#' 
#' @family CallSet functions
#' @export
CallSet <- function(CallSet.info = NULL, id = NULL, name = NULL, sampleId = NULL, 
    variantSetIds = NULL, created = NULL, info = NULL) {
    structure(list(CallSet.info = CallSet.info, id = id, name = name, sampleId = sampleId, 
        variantSetIds = variantSetIds, created = created, info = info), class = "gar_CallSet")
}

#' CallSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A map of additional call set information. This must be of the form map (string key mapping to a list of string values).
#' 
#' 
#' 
#' @return CallSet.info object
#' 
#' @family CallSet functions
#' @export
CallSet.info <- function() {
    list()
}

#' StreamVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The stream variants request.
#' 
#' @param projectId The Google Developers Console project ID or number which will be billed for this access
#' @param variantSetId The variant set ID from which to stream variants
#' @param callSetIds Only return variant calls which belong to call sets with these IDs
#' @param referenceName Required
#' @param start The beginning of the window (0-based, inclusive) for which overlapping variants should be returned
#' @param end The end of the window (0-based, exclusive) for which overlapping variants should be returned
#' 
#' @return StreamVariantsRequest object
#' 
#' @family StreamVariantsRequest functions
#' @export
StreamVariantsRequest <- function(projectId = NULL, variantSetId = NULL, callSetIds = NULL, 
    referenceName = NULL, start = NULL, end = NULL) {
    structure(list(projectId = projectId, variantSetId = variantSetId, callSetIds = callSetIds, 
        referenceName = referenceName, start = start, end = end), class = "gar_StreamVariantsRequest")
}

#' StreamVariantsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param variants No description
#' 
#' @return StreamVariantsResponse object
#' 
#' @family StreamVariantsResponse functions
#' @export
StreamVariantsResponse <- function(variants = NULL) {
    structure(list(variants = variants), class = "gar_StreamVariantsResponse")
}

#' ImportReadGroupSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set import response.
#' 
#' @param readGroupSetIds IDs of the read group sets that were created
#' 
#' @return ImportReadGroupSetsResponse object
#' 
#' @family ImportReadGroupSetsResponse functions
#' @export
ImportReadGroupSetsResponse <- function(readGroupSetIds = NULL) {
    structure(list(readGroupSetIds = readGroupSetIds), class = "gar_ImportReadGroupSetsResponse")
}

#' ImportVariantsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data import response.
#' 
#' @param callSetIds IDs of the call sets created during the import
#' 
#' @return ImportVariantsResponse object
#' 
#' @family ImportVariantsResponse functions
#' @export
ImportVariantsResponse <- function(callSetIds = NULL) {
    structure(list(callSetIds = callSetIds), class = "gar_ImportVariantsResponse")
}

#' OperationMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Metadata describing an Operation.
#' 
#' @param OperationMetadata.request The \link{OperationMetadata.request} object or list of objects
#' @param OperationMetadata.runtimeMetadata The \link{OperationMetadata.runtimeMetadata} object or list of objects
#' @param projectId The Google Cloud Project in which the job is scoped
#' @param createTime The time at which the job was submitted to the Genomics service
#' @param startTime The time at which the job began to run
#' @param endTime The time at which the job stopped running
#' @param request The original request that started the operation
#' @param events Optional event messages that were generated during the job's execution
#' @param clientId Optionally provided by the caller when submitting the request that creates the operation
#' @param runtimeMetadata Runtime metadata on this Operation
#' 
#' @return OperationMetadata object
#' 
#' @family OperationMetadata functions
#' @export
OperationMetadata <- function(OperationMetadata.request = NULL, OperationMetadata.runtimeMetadata = NULL, 
    projectId = NULL, createTime = NULL, startTime = NULL, endTime = NULL, request = NULL, 
    events = NULL, clientId = NULL, runtimeMetadata = NULL) {
    structure(list(OperationMetadata.request = OperationMetadata.request, OperationMetadata.runtimeMetadata = OperationMetadata.runtimeMetadata, 
        projectId = projectId, createTime = createTime, startTime = startTime, endTime = endTime, 
        request = request, events = events, clientId = clientId, runtimeMetadata = runtimeMetadata), 
        class = "gar_OperationMetadata")
}

#' OperationMetadata.request Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The original request that started the operation. Note that this will be in current version of the API. If the operation was started with v1beta2 API and a GetOperation is performed on v1 API, a v1 request will be returned.
#' 
#' 
#' 
#' @return OperationMetadata.request object
#' 
#' @family OperationMetadata functions
#' @export
OperationMetadata.request <- function() {
    list()
}

#' OperationMetadata.runtimeMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Runtime metadata on this Operation.
#' 
#' 
#' 
#' @return OperationMetadata.runtimeMetadata object
#' 
#' @family OperationMetadata functions
#' @export
OperationMetadata.runtimeMetadata <- function() {
    list()
}

#' OperationEvent Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An event that occurred during an Operation.
#' 
#' @param startTime Optional time of when event started
#' @param endTime Optional time of when event finished
#' @param description Required description of event
#' 
#' @return OperationEvent object
#' 
#' @family OperationEvent functions
#' @export
OperationEvent <- function(startTime = NULL, endTime = NULL, description = NULL) {
    structure(list(startTime = startTime, endTime = endTime, description = description), 
        class = "gar_OperationEvent")
}

#' RuntimeMetadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Runtime metadata that will be populated in the runtimeMetadata field of the Operation associated with a RunPipeline execution.
#' 
#' @param computeEngine Execution information specific to Google Compute Engine
#' 
#' @return RuntimeMetadata object
#' 
#' @family RuntimeMetadata functions
#' @export
RuntimeMetadata <- function(computeEngine = NULL) {
    structure(list(computeEngine = computeEngine), class = "gar_RuntimeMetadata")
}


#' ComputeEngine Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Describes a Compute Engine resource that is being managed by a running pipeline.
#' 
#' @param instanceName The instance on which the operation is running
#' @param zone The availability zone in which the instance resides
#' @param machineType The machine type of the instance
#' @param diskNames The names of the disks that were created for this pipeline
#' 
#' @return ComputeEngine object
#' 
#' @family ComputeEngine functions
#' @export


ComputeEngine <- function(instanceName = NULL, zone = NULL, machineType = NULL, diskNames = NULL) {
    
    
    
    structure(list(instanceName = instanceName, zone = zone, machineType = machineType, 
        diskNames = diskNames), class = "gar_ComputeEngine")
}

