#' Genomics API
#' Provides access to Genomics data.
#' 
#' Auto-generated code by googleAuthR::gar_create_api_skeleton
#'  at 2016-09-03 23:14:36
#' filename: /Users/mark/dev/R/autoGoogleAPI/googlegenomicsv1beta2.auto/R/genomics_functions.R
#' api_json: api_json
#' 
#' @details 
#' Authentication scopes used are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/bigquery
#' \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' @docType package 
#' @name genomics_googleAuthR
#' 
NULL
## NULL


#' Creates a new annotation set. Caller must have WRITE permission for the associated dataset.The following fields must be provided when creating an annotation set:  - datasetId - referenceSetId  All other fields may be optionally specified, unless documented as being server-generated (for example, the id field).
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param AnnotationSet The \link{AnnotationSet} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family AnnotationSet functions
#' @export
annotationSets.create <- function(AnnotationSet) {
    url <- "https://www.googleapis.com/genomics/v1beta2/annotationSets"
    # genomics.annotationSets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(AnnotationSet, "gar_AnnotationSet"))
    
    f(the_body = AnnotationSet)
    
}

#' Deletes an annotation set. Caller must have WRITE permission for the associated annotation set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationSetId The ID of the annotation set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
annotationSets.delete <- function(annotationSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotationSets/%s", 
        annotationSetId)
    # genomics.annotationSets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Gets an annotation set. Caller must have READ permission for the associated dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationSetId The ID of the annotation set to be retrieved
#' @importFrom googleAuthR gar_api_generator
#' @export
annotationSets.get <- function(annotationSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotationSets/%s", 
        annotationSetId)
    # genomics.annotationSets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Updates an annotation set. The update must respect all mutability restrictions and other invariants described on the annotation set resource. Caller must have WRITE permission for the associated dataset. This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param AnnotationSet The \link{AnnotationSet} object to pass to this method
#' @param annotationSetId The ID of the annotation set to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family AnnotationSet functions
#' @export
annotationSets.patch <- function(AnnotationSet, annotationSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotationSets/%s", 
        annotationSetId)
    # genomics.annotationSets.patch
    f <- gar_api_generator(url, "PATCH", data_parse_function = function(x) x)
    stopifnot(inherits(AnnotationSet, "gar_AnnotationSet"))
    
    f(the_body = AnnotationSet)
    
}

#' Searches for annotation sets that match the given criteria. Annotation sets are returned in an unspecified order. This order is consistent, such that two queries for the same content (regardless of page size) yield annotation sets in the same order across their respective streams of paginated responses. Caller must have READ permission for the queried datasets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchAnnotationSetsRequest The \link{SearchAnnotationSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchAnnotationSetsRequest functions
#' @export
annotationSets.search <- function(SearchAnnotationSetsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/annotationSets/search"
    # genomics.annotationSets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchAnnotationSetsRequest, "gar_SearchAnnotationSetsRequest"))
    
    f(the_body = SearchAnnotationSetsRequest)
    
}

#' Updates an annotation set. The update must respect all mutability restrictions and other invariants described on the annotation set resource. Caller must have WRITE permission for the associated dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param AnnotationSet The \link{AnnotationSet} object to pass to this method
#' @param annotationSetId The ID of the annotation set to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family AnnotationSet functions
#' @export
annotationSets.update <- function(AnnotationSet, annotationSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotationSets/%s", 
        annotationSetId)
    # genomics.annotationSets.update
    f <- gar_api_generator(url, "PUT", data_parse_function = function(x) x)
    stopifnot(inherits(AnnotationSet, "gar_AnnotationSet"))
    
    f(the_body = AnnotationSet)
    
}

#' Creates one or more new annotations atomically. All annotations must belong to the same annotation set. Caller must have WRITE permission for this annotation set. For optimal performance, batch positionally adjacent annotations together.If the request has a systemic issue, such as an attempt to write to an inaccessible annotation set, the entire RPC will fail accordingly. For lesser data issues, when possible an error will be isolated to the corresponding batch entry in the response; the remaining well formed annotations will be created normally.For details on the requirements for each individual annotation resource, see annotations.create.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param BatchCreateAnnotationsRequest The \link{BatchCreateAnnotationsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family BatchCreateAnnotationsRequest functions
#' @export
annotations.batchCreate <- function(BatchCreateAnnotationsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/annotations:batchCreate"
    # genomics.annotations.batchCreate
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(BatchCreateAnnotationsRequest, "gar_BatchCreateAnnotationsRequest"))
    
    f(the_body = BatchCreateAnnotationsRequest)
    
}

#' Creates a new annotation. Caller must have WRITE permission for the associated annotation set.The following fields must be provided when creating an annotation:  - annotationSetId - position.referenceName or  position.referenceId  Transcripts For annotations of type TRANSCRIPT, the following fields of annotation.transcript must be provided:  - exons.start - exons.end  All other fields may be optionally specified, unless documented as being server-generated (for example, the id field). The annotated range must be no longer than 100Mbp (mega base pairs). See the annotation resource for additional restrictions on each field.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Annotation The \link{Annotation} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family Annotation functions
#' @export
annotations.create <- function(Annotation) {
    url <- "https://www.googleapis.com/genomics/v1beta2/annotations"
    # genomics.annotations.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(Annotation, "gar_Annotation"))
    
    f(the_body = Annotation)
    
}

#' Deletes an annotation. Caller must have WRITE permission for the associated annotation set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationId The ID of the annotation to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
annotations.delete <- function(annotationId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotations/%s", 
        annotationId)
    # genomics.annotations.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Gets an annotation. Caller must have READ permission for the associated annotation set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param annotationId The ID of the annotation to be retrieved
#' @importFrom googleAuthR gar_api_generator
#' @export
annotations.get <- function(annotationId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotations/%s", 
        annotationId)
    # genomics.annotations.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Updates an annotation. The update must respect all mutability restrictions and other invariants described on the annotation resource. Caller must have WRITE permission for the associated dataset. This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Annotation The \link{Annotation} object to pass to this method
#' @param annotationId The ID of the annotation to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family Annotation functions
#' @export
annotations.patch <- function(Annotation, annotationId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotations/%s", 
        annotationId)
    # genomics.annotations.patch
    f <- gar_api_generator(url, "PATCH", data_parse_function = function(x) x)
    stopifnot(inherits(Annotation, "gar_Annotation"))
    
    f(the_body = Annotation)
    
}

#' Searches for annotations that match the given criteria. Results are ordered by genomic coordinate (by reference sequence, then position). Annotations with equivalent genomic coordinates are returned in an unspecified order. This order is consistent, such that two queries for the same content (regardless of page size) yield annotations in the same order across their respective streams of paginated responses. Caller must have READ permission for the queried annotation sets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchAnnotationsRequest The \link{SearchAnnotationsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchAnnotationsRequest functions
#' @export
annotations.search <- function(SearchAnnotationsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/annotations/search"
    # genomics.annotations.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchAnnotationsRequest, "gar_SearchAnnotationsRequest"))
    
    f(the_body = SearchAnnotationsRequest)
    
}

#' Updates an annotation. The update must respect all mutability restrictions and other invariants described on the annotation resource. Caller must have WRITE permission for the associated dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Annotation The \link{Annotation} object to pass to this method
#' @param annotationId The ID of the annotation to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family Annotation functions
#' @export
annotations.update <- function(Annotation, annotationId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/annotations/%s", 
        annotationId)
    # genomics.annotations.update
    f <- gar_api_generator(url, "PUT", data_parse_function = function(x) x)
    stopifnot(inherits(Annotation, "gar_Annotation"))
    
    f(the_body = Annotation)
    
}

#' Creates a new call set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param CallSet The \link{CallSet} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family CallSet functions
#' @export
callsets.create <- function(CallSet) {
    url <- "https://www.googleapis.com/genomics/v1beta2/callsets"
    # genomics.callsets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(CallSet, "gar_CallSet"))
    
    f(the_body = CallSet)
    
}

#' Deletes a call set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param callSetId The ID of the call set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
callsets.delete <- function(callSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/callsets/%s", callSetId)
    # genomics.callsets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Gets a call set by ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param callSetId The ID of the call set
#' @importFrom googleAuthR gar_api_generator
#' @export
callsets.get <- function(callSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/callsets/%s", callSetId)
    # genomics.callsets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Updates a call set. This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param CallSet The \link{CallSet} object to pass to this method
#' @param callSetId Required
#' @importFrom googleAuthR gar_api_generator
#' @family CallSet functions
#' @export
callsets.patch <- function(CallSet, callSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/callsets/%s", callSetId)
    # genomics.callsets.patch
    f <- gar_api_generator(url, "PATCH", data_parse_function = function(x) x)
    stopifnot(inherits(CallSet, "gar_CallSet"))
    
    f(the_body = CallSet)
    
}

#' Gets a list of call sets matching the criteria.Implements GlobalAllianceApi.searchCallSets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchCallSetsRequest The \link{SearchCallSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchCallSetsRequest functions
#' @export
callsets.search <- function(SearchCallSetsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/callsets/search"
    # genomics.callsets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchCallSetsRequest, "gar_SearchCallSetsRequest"))
    
    f(the_body = SearchCallSetsRequest)
    
}

#' Updates a call set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param CallSet The \link{CallSet} object to pass to this method
#' @param callSetId Required
#' @importFrom googleAuthR gar_api_generator
#' @family CallSet functions
#' @export
callsets.update <- function(CallSet, callSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/callsets/%s", callSetId)
    # genomics.callsets.update
    f <- gar_api_generator(url, "PUT", data_parse_function = function(x) x)
    stopifnot(inherits(CallSet, "gar_CallSet"))
    
    f(the_body = CallSet)
    
}

#' Creates a new dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Dataset The \link{Dataset} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family Dataset functions
#' @export
datasets.create <- function(Dataset) {
    url <- "https://www.googleapis.com/genomics/v1beta2/datasets"
    # genomics.datasets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(Dataset, "gar_Dataset"))
    
    f(the_body = Dataset)
    
}

#' Deletes a dataset and all of its contents (all read group sets, reference sets, variant sets, call sets, annotation sets, etc.) This is reversible (up to one week after the deletion) via the UndeleteDataset operation.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param datasetId The ID of the dataset to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
datasets.delete <- function(datasetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/datasets/%s", datasetId)
    # genomics.datasets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Gets a dataset by ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param datasetId The ID of the dataset
#' @importFrom googleAuthR gar_api_generator
#' @export
datasets.get <- function(datasetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/datasets/%s", datasetId)
    # genomics.datasets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Lists datasets within a project.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param projectNumber Required
#' @importFrom googleAuthR gar_api_generator
#' @export
datasets.list <- function(pageSize = NULL, pageToken = NULL, projectNumber = NULL) {
    url <- "https://www.googleapis.com/genomics/v1beta2/datasets"
    # genomics.datasets.list
    f <- gar_api_generator(url, "GET", pars_args = list(pageSize = pageSize, pageToken = pageToken, 
        projectNumber = projectNumber), data_parse_function = function(x) x)
    f()
    
}

#' Updates a dataset. This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Dataset The \link{Dataset} object to pass to this method
#' @param datasetId The ID of the dataset to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family Dataset functions
#' @export
datasets.patch <- function(Dataset, datasetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/datasets/%s", datasetId)
    # genomics.datasets.patch
    f <- gar_api_generator(url, "PATCH", data_parse_function = function(x) x)
    stopifnot(inherits(Dataset, "gar_Dataset"))
    
    f(the_body = Dataset)
    
}

#' Undeletes a dataset by restoring a dataset which was deleted via this API. This operation is only possible for a week after the deletion occurred.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param datasetId The ID of the dataset to be undeleted
#' @importFrom googleAuthR gar_api_generator
#' @export
datasets.undelete <- function(datasetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/datasets/%s/undelete", 
        datasetId)
    # genomics.datasets.undelete
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    f()
    
}

#' Updates a dataset.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Dataset The \link{Dataset} object to pass to this method
#' @param datasetId The ID of the dataset to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family Dataset functions
#' @export
datasets.update <- function(Dataset, datasetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/datasets/%s", datasetId)
    # genomics.datasets.update
    f <- gar_api_generator(url, "PUT", data_parse_function = function(x) x)
    stopifnot(inherits(Dataset, "gar_Dataset"))
    
    f(the_body = Dataset)
    
}

#' Creates and asynchronously runs an ad-hoc job. This is an experimental call and may be removed or changed at any time.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/devstorage.read_write, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ExperimentalCreateJobRequest The \link{ExperimentalCreateJobRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family ExperimentalCreateJobRequest functions
#' @export
experimental.jobs.create <- function(ExperimentalCreateJobRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/experimental/jobs/create"
    # genomics.experimental.jobs.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ExperimentalCreateJobRequest, "gar_ExperimentalCreateJobRequest"))
    
    f(the_body = ExperimentalCreateJobRequest)
    
}

#' Cancels a job by ID. Note that it is possible for partial results to be generated and stored for cancelled jobs.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param jobId Required
#' @importFrom googleAuthR gar_api_generator
#' @export
jobs.cancel <- function(jobId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/jobs/%s/cancel", 
        jobId)
    # genomics.jobs.cancel
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    f()
    
}

#' Gets a job by ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param jobId Required
#' @importFrom googleAuthR gar_api_generator
#' @export
jobs.get <- function(jobId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/jobs/%s", jobId)
    # genomics.jobs.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Gets a list of jobs matching the criteria.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchJobsRequest The \link{SearchJobsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchJobsRequest functions
#' @export
jobs.search <- function(SearchJobsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/jobs/search"
    # genomics.jobs.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchJobsRequest, "gar_SearchJobsRequest"))
    
    f(the_body = SearchJobsRequest)
    
}

#' Deletes a read group set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param readGroupSetId The ID of the read group set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
readgroupsets.delete <- function(readGroupSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/readgroupsets/%s", 
        readGroupSetId)
    # genomics.readgroupsets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Exports read group sets to a BAM file in Google Cloud Storage.Note that currently there may be some differences between exported BAM files and the original BAM file at the time of import. See ImportReadGroupSets for details.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/devstorage.read_write, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ExportReadGroupSetsRequest The \link{ExportReadGroupSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family ExportReadGroupSetsRequest functions
#' @export
readgroupsets.export <- function(ExportReadGroupSetsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/readgroupsets/export"
    # genomics.readgroupsets.export
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ExportReadGroupSetsRequest, "gar_ExportReadGroupSetsRequest"))
    
    f(the_body = ExportReadGroupSetsRequest)
    
}

#' Gets a read group set by ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param readGroupSetId The ID of the read group set
#' @importFrom googleAuthR gar_api_generator
#' @export
readgroupsets.get <- function(readGroupSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/readgroupsets/%s", 
        readGroupSetId)
    # genomics.readgroupsets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Creates read group sets by asynchronously importing the provided information. The caller must have WRITE permissions to the dataset.Notes on BAM import:  - Tags will be converted to strings - tag types are not preserved- Comments (@CO) in the input file header are not imported- Original order of reference headers is not preserved- Any reverse stranded unmapped reads will be reverse complemented, and their qualities (also the 'BQ' and 'OQ' tags, if any) will be reversed- Unmapped reads will be stripped of positional information (referenceName and position)
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/devstorage.read_write, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ImportReadGroupSetsRequest The \link{ImportReadGroupSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family ImportReadGroupSetsRequest functions
#' @export
readgroupsets.import <- function(ImportReadGroupSetsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/readgroupsets/import"
    # genomics.readgroupsets.import
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ImportReadGroupSetsRequest, "gar_ImportReadGroupSetsRequest"))
    
    f(the_body = ImportReadGroupSetsRequest)
    
}

#' Updates a read group set. This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ReadGroupSet The \link{ReadGroupSet} object to pass to this method
#' @param readGroupSetId The ID of the read group set to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family ReadGroupSet functions
#' @export
readgroupsets.patch <- function(ReadGroupSet, readGroupSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/readgroupsets/%s", 
        readGroupSetId)
    # genomics.readgroupsets.patch
    f <- gar_api_generator(url, "PATCH", data_parse_function = function(x) x)
    stopifnot(inherits(ReadGroupSet, "gar_ReadGroupSet"))
    
    f(the_body = ReadGroupSet)
    
}

#' Searches for read group sets matching the criteria.Implements GlobalAllianceApi.searchReadGroupSets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReadGroupSetsRequest The \link{SearchReadGroupSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReadGroupSetsRequest functions
#' @export
readgroupsets.search <- function(SearchReadGroupSetsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/readgroupsets/search"
    # genomics.readgroupsets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReadGroupSetsRequest, "gar_SearchReadGroupSetsRequest"))
    
    f(the_body = SearchReadGroupSetsRequest)
    
}

#' Updates a read group set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ReadGroupSet The \link{ReadGroupSet} object to pass to this method
#' @param readGroupSetId The ID of the read group set to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family ReadGroupSet functions
#' @export
readgroupsets.update <- function(ReadGroupSet, readGroupSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/readgroupsets/%s", 
        readGroupSetId)
    # genomics.readgroupsets.update
    f <- gar_api_generator(url, "PUT", data_parse_function = function(x) x)
    stopifnot(inherits(ReadGroupSet, "gar_ReadGroupSet"))
    
    f(the_body = ReadGroupSet)
    
}

#' Gets a list of reads for one or more read group sets. Reads search operates over a genomic coordinate space of reference sequence & position defined over the reference sequences to which the requested read group sets are aligned.If a target positional range is specified, search returns all reads whose alignment to the reference genome overlap the range. A query which specifies only read group set IDs yields all reads in those read group sets, including unmapped reads.All reads returned (including reads on subsequent pages) are ordered by genomic coordinate (by reference sequence, then position). Reads with equivalent genomic coordinates are returned in an unspecified order. This order is consistent, such that two queries for the same content (regardless of page size) yield reads in the same order across their respective streams of paginated responses.Implements GlobalAllianceApi.searchReads.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReadsRequest The \link{SearchReadsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReadsRequest functions
#' @export
reads.search <- function(SearchReadsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/reads/search"
    # genomics.reads.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReadsRequest, "gar_SearchReadsRequest"))
    
    f(the_body = SearchReadsRequest)
    
}

#' Gets a reference.Implements GlobalAllianceApi.getReference.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param referenceId The ID of the reference
#' @importFrom googleAuthR gar_api_generator
#' @export
references.get <- function(referenceId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/references/%s", referenceId)
    # genomics.references.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Searches for references which match the given criteria.Implements GlobalAllianceApi.searchReferences.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReferencesRequest The \link{SearchReferencesRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReferencesRequest functions
#' @export
references.search <- function(SearchReferencesRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/references/search"
    # genomics.references.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReferencesRequest, "gar_SearchReferencesRequest"))
    
    f(the_body = SearchReferencesRequest)
    
}

#' Gets a reference set.Implements GlobalAllianceApi.getReferenceSet.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param referenceSetId The ID of the reference set
#' @importFrom googleAuthR gar_api_generator
#' @export
referencesets.get <- function(referenceSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/referencesets/%s", 
        referenceSetId)
    # genomics.referencesets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Searches for reference sets which match the given criteria.Implements GlobalAllianceApi.searchReferenceSets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchReferenceSetsRequest The \link{SearchReferenceSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchReferenceSetsRequest functions
#' @export
referencesets.search <- function(SearchReferenceSetsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/referencesets/search"
    # genomics.referencesets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchReferenceSetsRequest, "gar_SearchReferenceSetsRequest"))
    
    f(the_body = SearchReferenceSetsRequest)
    
}

#' Creates a new variant.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Variant The \link{Variant} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family Variant functions
#' @export
variants.create <- function(Variant) {
    url <- "https://www.googleapis.com/genomics/v1beta2/variants"
    # genomics.variants.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(Variant, "gar_Variant"))
    
    f(the_body = Variant)
    
}

#' Deletes a variant.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantId The ID of the variant to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
variants.delete <- function(variantId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variants/%s", variantId)
    # genomics.variants.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Gets a variant by ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantId Required
#' @importFrom googleAuthR gar_api_generator
#' @export
variants.get <- function(variantId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variants/%s", variantId)
    # genomics.variants.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Gets a list of variants matching the criteria.Implements GlobalAllianceApi.searchVariants.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchVariantsRequest The \link{SearchVariantsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchVariantsRequest functions
#' @export
variants.search <- function(SearchVariantsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/variants/search"
    # genomics.variants.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchVariantsRequest, "gar_SearchVariantsRequest"))
    
    f(the_body = SearchVariantsRequest)
    
}

#' Updates a variant's names and info fields. All other modifications are silently ignored. Returns the modified variant without its calls.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param Variant The \link{Variant} object to pass to this method
#' @param variantId The ID of the variant to be updated
#' @importFrom googleAuthR gar_api_generator
#' @family Variant functions
#' @export
variants.update <- function(Variant, variantId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variants/%s", variantId)
    # genomics.variants.update
    f <- gar_api_generator(url, "PUT", data_parse_function = function(x) x)
    stopifnot(inherits(Variant, "gar_Variant"))
    
    f(the_body = Variant)
    
}

#' Creates a new variant set (only necessary in v1).The provided variant set must have a valid datasetId set - all other fields are optional. Note that the id field will be ignored, as this is assigned by the server.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param VariantSet The \link{VariantSet} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family VariantSet functions
#' @export
variantsets.create <- function(VariantSet) {
    url <- "https://www.googleapis.com/genomics/v1beta2/variantsets"
    # genomics.variantsets.create
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(VariantSet, "gar_VariantSet"))
    
    f(the_body = VariantSet)
    
}

#' Deletes a variant set including all variants, call sets, and calls within. This is not reversible.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantSetId The ID of the variant set to be deleted
#' @importFrom googleAuthR gar_api_generator
#' @export
variantsets.delete <- function(variantSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variantsets/%s", 
        variantSetId)
    # genomics.variantsets.delete
    f <- gar_api_generator(url, "DELETE", data_parse_function = function(x) x)
    f()
    
}

#' Exports variant set data to an external destination.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/bigquery
#' \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/bigquery, https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ExportVariantSetRequest The \link{ExportVariantSetRequest} object to pass to this method
#' @param variantSetId The ID of the variant set that contains variant data which should be exported
#' @importFrom googleAuthR gar_api_generator
#' @family ExportVariantSetRequest functions
#' @export
variantsets.export <- function(ExportVariantSetRequest, variantSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variantsets/%s/export", 
        variantSetId)
    # genomics.variantsets.export
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ExportVariantSetRequest, "gar_ExportVariantSetRequest"))
    
    f(the_body = ExportVariantSetRequest)
    
}

#' Gets a variant set by ID.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param variantSetId The ID of the variant set
#' @importFrom googleAuthR gar_api_generator
#' @export
variantsets.get <- function(variantSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variantsets/%s", 
        variantSetId)
    # genomics.variantsets.get
    f <- gar_api_generator(url, "GET", data_parse_function = function(x) x)
    f()
    
}

#' Creates variant data by asynchronously importing the provided information.The variants for import will be merged with any existing data and each other according to the behavior of mergeVariants. In particular, this means for merged VCF variants that have conflicting info fields, some data will be arbitrarily discarded unless otherwise specified in the InfoMergeConfig field of ImportVariantsRequest. As a special case, for single-sample VCF files, QUAL and FILTER fields will be moved to the call level; these are sometimes interpreted in a call-specific context. Imported VCF headers are appended to the metadata already in a variant set.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/devstorage.read_write
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/devstorage.read_write, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param ImportVariantsRequest The \link{ImportVariantsRequest} object to pass to this method
#' @param variantSetId Required
#' @importFrom googleAuthR gar_api_generator
#' @family ImportVariantsRequest functions
#' @export
variantsets.importVariants <- function(ImportVariantsRequest, variantSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variantsets/%s/importVariants", 
        variantSetId)
    # genomics.variantsets.importVariants
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(ImportVariantsRequest, "gar_ImportVariantsRequest"))
    
    f(the_body = ImportVariantsRequest)
    
}

#' Merges the given variants with existing variants. Each variant will be merged with an existing variant that matches its reference sequence, start, end, reference bases, and alternative bases. If no such variant exists, a new one will be created.When variants are merged, the call information from the new variant is added to the existing variant. Variant info fields are merged as specified in the InfoMergeConfig field of the MergeVariantsRequest.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param MergeVariantsRequest The \link{MergeVariantsRequest} object to pass to this method
#' @param variantSetId The destination variant set
#' @importFrom googleAuthR gar_api_generator
#' @family MergeVariantsRequest functions
#' @export
variantsets.mergeVariants <- function(MergeVariantsRequest, variantSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variantsets/%s/mergeVariants", 
        variantSetId)
    # genomics.variantsets.mergeVariants
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(MergeVariantsRequest, "gar_MergeVariantsRequest"))
    
    f(the_body = MergeVariantsRequest)
    
}

#' Updates a variant set's metadata. All other modifications are silently ignored. This method supports patch semantics.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param VariantSet The \link{VariantSet} object to pass to this method
#' @param variantSetId The ID of the variant set to be updated (must already exist)
#' @importFrom googleAuthR gar_api_generator
#' @family VariantSet functions
#' @export
variantsets.patch <- function(VariantSet, variantSetId) {
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variantsets/%s", 
        variantSetId)
    # genomics.variantsets.patch
    f <- gar_api_generator(url, "PATCH", data_parse_function = function(x) x)
    stopifnot(inherits(VariantSet, "gar_VariantSet"))
    
    f(the_body = VariantSet)
    
}

#' Returns a list of all variant sets matching search criteria.Implements GlobalAllianceApi.searchVariantSets.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' \item https://www.googleapis.com/auth/genomics.readonly
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics, https://www.googleapis.com/auth/genomics.readonly)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param SearchVariantSetsRequest The \link{SearchVariantSetsRequest} object to pass to this method
#' #' @importFrom googleAuthR gar_api_generator
#' @family SearchVariantSetsRequest functions
#' @export
variantsets.search <- function(SearchVariantSetsRequest) {
    url <- "https://www.googleapis.com/genomics/v1beta2/variantsets/search"
    # genomics.variantsets.search
    f <- gar_api_generator(url, "POST", data_parse_function = function(x) x)
    stopifnot(inherits(SearchVariantSetsRequest, "gar_SearchVariantSetsRequest"))
    
    f(the_body = SearchVariantSetsRequest)
    
}


#' Updates a variant set's metadata. All other modifications are silently ignored.
#' 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_skeleton}}
#' 
#' @seealso \href{https://developers.google.com/genomics/v1beta2/reference}{Google Documentation}
#' 
#' @details 
#' Authentication scopes used by this function are:
#' \itemize{
#'   \item https://www.googleapis.com/auth/cloud-platform
#' \item https://www.googleapis.com/auth/genomics
#' }
#' 
#' Set \code{options(googleAuthR.scopes.selected = c(https://www.googleapis.com/auth/cloud-platform, https://www.googleapis.com/auth/genomics)}
#' Then run \code{googleAuthR::gar_auth()} to authenticate.
#' See \code{\link[googleAuthR]{gar_auth}} for details. 
#' 
#' @param VariantSet The \link{VariantSet} object to pass to this method
#' @param variantSetId The ID of the variant set to be updated (must already exist)
#' @importFrom googleAuthR gar_api_generator
#' @family VariantSet functions
#' @export


variantsets.update <- function(VariantSet, variantSetId) {
    
    
    url <- sprintf("https://www.googleapis.com/genomics/v1beta2/variantsets/%s", 
        variantSetId)
    # genomics.variantsets.update
    f <- gar_api_generator(url, "PUT", data_parse_function = function(x) x)
    
    stopifnot(inherits(VariantSet, "gar_VariantSet"))
    
    f(the_body = VariantSet)
    
    
}



