#' Genomics API Objects 
#' Provides access to Genomics data.
#' 
#' Auto-generated code by googleAuthR::gar_create_api_objects
#'  at 2016-09-03 23:14:37
#' filename: /Users/mark/dev/R/autoGoogleAPI/googlegenomicsv1beta2.auto/R/genomics_objects.R
#' api_json: api_json
#' 
#' Objects for use by the functions created by googleAuthR::gar_create_api_skeleton

#' Annotation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An annotation describes a region of reference genome. The value of an annotation may be one of several canonical types, supplemented by arbitrary info tags. An annotation is not inherently associated with a specific sample or individual (though a client could choose to use annotations in this way). Example canonical annotation types are GENE and VARIANT.
#' 
#' @param Annotation.info The \link{Annotation.info} object or list of objects
#' @param annotationSetId The annotation set to which this annotation belongs
#' @param id The server-generated annotation ID, unique across all annotations
#' @param info A string which maps to an array of values
#' @param name The display name of this annotation
#' @param position The position of this annotation on the reference sequence
#' @param transcript A transcript value represents the assertion that a particular region of the reference genome may be transcribed as RNA
#' @param type The data type for this annotation
#' @param variant A variant annotation, which describes the effect of a variant on the genome, the coding sequence, and/or higher level consequences at the organism level e
#' 
#' @return Annotation object
#' 
#' @family Annotation functions
#' @export
Annotation <- function(Annotation.info = NULL, annotationSetId = NULL, id = NULL, 
    info = NULL, name = NULL, position = NULL, transcript = NULL, type = NULL, variant = NULL) {
    structure(list(Annotation.info = Annotation.info, annotationSetId = annotationSetId, 
        id = id, info = info, name = name, position = position, transcript = transcript, 
        type = type, variant = variant), class = "gar_Annotation")
}

#' Annotation.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return Annotation.info object
#' 
#' @family Annotation functions
#' @export
Annotation.info <- function() {
    list()
}

#' AnnotationSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An annotation set is a logical grouping of annotations that share consistent type information and provenance. Examples of annotation sets include 'all genes from refseq', and 'all variant annotations from ClinVar'.
#' 
#' @param AnnotationSet.info The \link{AnnotationSet.info} object or list of objects
#' @param datasetId The dataset to which this annotation set belongs
#' @param id The server-generated annotation set ID, unique across all annotation sets
#' @param info A string which maps to an array of values
#' @param name The display name for this annotation set
#' @param referenceSetId The ID of the reference set that defines the coordinate space for this set's annotations
#' @param sourceUri The source URI describing the file from which this annotation set was generated, if any
#' @param type The type of annotations contained within this set
#' 
#' @return AnnotationSet object
#' 
#' @family AnnotationSet functions
#' @export
AnnotationSet <- function(AnnotationSet.info = NULL, datasetId = NULL, id = NULL, 
    info = NULL, name = NULL, referenceSetId = NULL, sourceUri = NULL, type = NULL) {
    structure(list(AnnotationSet.info = AnnotationSet.info, datasetId = datasetId, 
        id = id, info = info, name = name, referenceSetId = referenceSetId, sourceUri = sourceUri, 
        type = type), class = "gar_AnnotationSet")
}

#' AnnotationSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return AnnotationSet.info object
#' 
#' @family AnnotationSet functions
#' @export
AnnotationSet.info <- function() {
    list()
}

#' BatchAnnotationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param entries The resulting per-annotation entries, ordered consistently with the original request
#' 
#' @return BatchAnnotationsResponse object
#' 
#' @family BatchAnnotationsResponse functions
#' @export
BatchAnnotationsResponse <- function(entries = NULL) {
    structure(list(entries = entries), class = "gar_BatchAnnotationsResponse")
}

#' BatchAnnotationsResponseEntry Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotation The annotation, if any
#' @param status The resulting status for this annotation operation
#' 
#' @return BatchAnnotationsResponseEntry object
#' 
#' @family BatchAnnotationsResponseEntry functions
#' @export
BatchAnnotationsResponseEntry <- function(annotation = NULL, status = NULL) {
    structure(list(annotation = annotation, status = status), class = "gar_BatchAnnotationsResponseEntry")
}

#' BatchAnnotationsResponseEntryStatus Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' 
#' 
#' @param code The HTTP status code for this operation
#' @param message Error message for this status, if any
#' 
#' @return BatchAnnotationsResponseEntryStatus object
#' 
#' @family BatchAnnotationsResponseEntryStatus functions
#' @export
BatchAnnotationsResponseEntryStatus <- function(code = NULL, message = NULL) {
    structure(list(code = code, message = message), class = "gar_BatchAnnotationsResponseEntryStatus")
}

#' BatchCreateAnnotationsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotations The annotations to be created
#' 
#' @return BatchCreateAnnotationsRequest object
#' 
#' @family BatchCreateAnnotationsRequest functions
#' @export
BatchCreateAnnotationsRequest <- function(annotations = NULL) {
    structure(list(annotations = annotations), class = "gar_BatchCreateAnnotationsRequest")
}

#' Call Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A call represents the determination of genotype with respect to a particular variant. It may include associated information such as quality and phasing. For example, a call might assign a probability of 0.32 to the occurrence of a SNP named rs1234 in a call set with the name NA12345.
#' 
#' @param Call.info The \link{Call.info} object or list of objects
#' @param callSetId The ID of the call set this variant call belongs to
#' @param callSetName The name of the call set this variant call belongs to
#' @param genotype The genotype of this variant call
#' @param genotypeLikelihood The genotype likelihoods for this variant call
#' @param info A string which maps to an array of values
#' @param phaseset If this field is present, this variant call's genotype ordering implies the phase of the bases and is consistent with any other variant calls in the same reference sequence which have the same phaseset value
#' 
#' @return Call object
#' 
#' @family Call functions
#' @export
Call <- function(Call.info = NULL, callSetId = NULL, callSetName = NULL, genotype = NULL, 
    genotypeLikelihood = NULL, info = NULL, phaseset = NULL) {
    structure(list(Call.info = Call.info, callSetId = callSetId, callSetName = callSetName, 
        genotype = genotype, genotypeLikelihood = genotypeLikelihood, info = info, 
        phaseset = phaseset), class = "gar_Call")
}

#' Call.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return Call.info object
#' 
#' @family Call functions
#' @export
Call.info <- function() {
    list()
}

#' CallSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A call set is a collection of variant calls, typically for one sample. It belongs to a variant set.
#' 
#' @param CallSet.info The \link{CallSet.info} object or list of objects
#' @param created The date this call set was created in milliseconds from the epoch
#' @param id The Google generated ID of the call set, immutable
#' @param info A string which maps to an array of values
#' @param name The call set name
#' @param sampleId The sample ID this call set corresponds to
#' @param variantSetIds The IDs of the variant sets this call set belongs to
#' 
#' @return CallSet object
#' 
#' @family CallSet functions
#' @export
CallSet <- function(CallSet.info = NULL, created = NULL, id = NULL, info = NULL, 
    name = NULL, sampleId = NULL, variantSetIds = NULL) {
    structure(list(CallSet.info = CallSet.info, created = created, id = id, info = info, 
        name = name, sampleId = sampleId, variantSetIds = variantSetIds), class = "gar_CallSet")
}

#' CallSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return CallSet.info object
#' 
#' @family CallSet functions
#' @export
CallSet.info <- function() {
    list()
}

#' CigarUnit Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A single CIGAR operation.
#' 
#' @param operation No description
#' @param operationLength The number of bases that the operation runs for
#' @param referenceSequence referenceSequence is only used at mismatches (SEQUENCE_MISMATCH) and deletions (DELETE)
#' 
#' @return CigarUnit object
#' 
#' @family CigarUnit functions
#' @export
CigarUnit <- function(operation = NULL, operationLength = NULL, referenceSequence = NULL) {
    structure(list(operation = operation, operationLength = operationLength, referenceSequence = referenceSequence), 
        class = "gar_CigarUnit")
}

#' CoverageBucket Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A bucket over which read coverage has been precomputed. A bucket corresponds to a specific range of the reference sequence.
#' 
#' @param meanCoverage The average number of reads which are aligned to each individual reference base in this bucket
#' @param range The genomic coordinate range spanned by this bucket
#' 
#' @return CoverageBucket object
#' 
#' @family CoverageBucket functions
#' @export
CoverageBucket <- function(meanCoverage = NULL, range = NULL) {
    structure(list(meanCoverage = meanCoverage, range = range), class = "gar_CoverageBucket")
}

#' Dataset Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A Dataset is a collection of genomic data.
#' 
#' @param createTime The time this dataset was created, in seconds from the epoch
#' @param id The Google generated ID of the dataset, immutable
#' @param isPublic Flag indicating whether or not a dataset is publicly viewable
#' @param name The dataset name
#' @param projectNumber The Google Developers Console project number that this dataset belongs to
#' 
#' @return Dataset object
#' 
#' @family Dataset functions
#' @export
Dataset <- function(createTime = NULL, id = NULL, isPublic = NULL, name = NULL, projectNumber = NULL) {
    structure(list(createTime = createTime, id = id, isPublic = isPublic, name = name, 
        projectNumber = projectNumber), class = "gar_Dataset")
}

#' ExperimentalCreateJobRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The job creation request.
#' 
#' @param align Specifies whether or not to run the alignment pipeline
#' @param callVariants Specifies whether or not to run the variant calling pipeline
#' @param gcsOutputPath Specifies where to copy the results of certain pipelines
#' @param pairedSourceUris A list of Google Cloud Storage URIs of paired end 
#' @param projectNumber Required
#' @param sourceUris A list of Google Cloud Storage URIs of data files to operate upon
#' 
#' @return ExperimentalCreateJobRequest object
#' 
#' @family ExperimentalCreateJobRequest functions
#' @export
ExperimentalCreateJobRequest <- function(align = NULL, callVariants = NULL, gcsOutputPath = NULL, 
    pairedSourceUris = NULL, projectNumber = NULL, sourceUris = NULL) {
    structure(list(align = align, callVariants = callVariants, gcsOutputPath = gcsOutputPath, 
        pairedSourceUris = pairedSourceUris, projectNumber = projectNumber, sourceUris = sourceUris), 
        class = "gar_ExperimentalCreateJobRequest")
}

#' ExperimentalCreateJobResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The job creation response.
#' 
#' @param jobId A job ID that can be used to get status information
#' 
#' @return ExperimentalCreateJobResponse object
#' 
#' @family ExperimentalCreateJobResponse functions
#' @export
ExperimentalCreateJobResponse <- function(jobId = NULL) {
    structure(list(jobId = jobId), class = "gar_ExperimentalCreateJobResponse")
}

#' ExportReadGroupSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set export request.
#' 
#' @param exportUri Required
#' @param projectNumber Required
#' @param readGroupSetIds Required
#' @param referenceNames The reference names to export
#' 
#' @return ExportReadGroupSetsRequest object
#' 
#' @family ExportReadGroupSetsRequest functions
#' @export
ExportReadGroupSetsRequest <- function(exportUri = NULL, projectNumber = NULL, readGroupSetIds = NULL, 
    referenceNames = NULL) {
    structure(list(exportUri = exportUri, projectNumber = projectNumber, readGroupSetIds = readGroupSetIds, 
        referenceNames = referenceNames), class = "gar_ExportReadGroupSetsRequest")
}

#' ExportReadGroupSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set export response.
#' 
#' @param jobId A job ID that can be used to get status information
#' 
#' @return ExportReadGroupSetsResponse object
#' 
#' @family ExportReadGroupSetsResponse functions
#' @export
ExportReadGroupSetsResponse <- function(jobId = NULL) {
    structure(list(jobId = jobId), class = "gar_ExportReadGroupSetsResponse")
}

#' ExportVariantSetRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data export request.
#' 
#' @param bigqueryDataset Required
#' @param bigqueryTable Required
#' @param callSetIds If provided, only variant call information from the specified call sets will be exported
#' @param format The format for the exported data
#' @param projectNumber Required
#' 
#' @return ExportVariantSetRequest object
#' 
#' @family ExportVariantSetRequest functions
#' @export
ExportVariantSetRequest <- function(bigqueryDataset = NULL, bigqueryTable = NULL, 
    callSetIds = NULL, format = NULL, projectNumber = NULL) {
    structure(list(bigqueryDataset = bigqueryDataset, bigqueryTable = bigqueryTable, 
        callSetIds = callSetIds, format = format, projectNumber = projectNumber), 
        class = "gar_ExportVariantSetRequest")
}

#' ExportVariantSetResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data export response.
#' 
#' @param jobId A job ID that can be used to get status information
#' 
#' @return ExportVariantSetResponse object
#' 
#' @family ExportVariantSetResponse functions
#' @export
ExportVariantSetResponse <- function(jobId = NULL) {
    structure(list(jobId = jobId), class = "gar_ExportVariantSetResponse")
}

#' ExternalId Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param id The id used by the source of this data
#' @param sourceName The name of the source of this data
#' 
#' @return ExternalId object
#' 
#' @family ExternalId functions
#' @export
ExternalId <- function(id = NULL, sourceName = NULL) {
    structure(list(id = id, sourceName = sourceName), class = "gar_ExternalId")
}

#' ImportReadGroupSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set import request.
#' 
#' @param datasetId Required
#' @param partitionStrategy The partition strategy describes how read groups are partitioned into read group sets
#' @param referenceSetId The reference set to which the imported read group sets are aligned to, if any
#' @param sourceUris A list of URIs pointing at BAM files in Google Cloud Storage
#' 
#' @return ImportReadGroupSetsRequest object
#' 
#' @family ImportReadGroupSetsRequest functions
#' @export
ImportReadGroupSetsRequest <- function(datasetId = NULL, partitionStrategy = NULL, 
    referenceSetId = NULL, sourceUris = NULL) {
    structure(list(datasetId = datasetId, partitionStrategy = partitionStrategy, 
        referenceSetId = referenceSetId, sourceUris = sourceUris), class = "gar_ImportReadGroupSetsRequest")
}

#' ImportReadGroupSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set import response.
#' 
#' @param jobId A job ID that can be used to get status information
#' 
#' @return ImportReadGroupSetsResponse object
#' 
#' @family ImportReadGroupSetsResponse functions
#' @export
ImportReadGroupSetsResponse <- function(jobId = NULL) {
    structure(list(jobId = jobId), class = "gar_ImportReadGroupSetsResponse")
}

#' ImportVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data import request.
#' 
#' @param format The format of the variant data being imported
#' @param normalizeReferenceNames Convert reference names to the canonical representation
#' @param sourceUris A list of URIs referencing variant files in Google Cloud Storage
#' 
#' @return ImportVariantsRequest object
#' 
#' @family ImportVariantsRequest functions
#' @export
ImportVariantsRequest <- function(format = NULL, normalizeReferenceNames = NULL, 
    sourceUris = NULL) {
    structure(list(format = format, normalizeReferenceNames = normalizeReferenceNames, 
        sourceUris = sourceUris), class = "gar_ImportVariantsRequest")
}

#' ImportVariantsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant data import response.
#' 
#' @param jobId A job ID that can be used to get status information
#' 
#' @return ImportVariantsResponse object
#' 
#' @family ImportVariantsResponse functions
#' @export
ImportVariantsResponse <- function(jobId = NULL) {
    structure(list(jobId = jobId), class = "gar_ImportVariantsResponse")
}

#' Int32Value Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Wrapper message for `int32`.The JSON representation for `Int32Value` is JSON number.
#' 
#' @param value The int32 value
#' 
#' @return Int32Value object
#' 
#' @family Int32Value functions
#' @export
Int32Value <- function(value = NULL) {
    structure(list(value = value), class = "gar_Int32Value")
}

#' Job Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A Job represents an ongoing process that can be monitored for status information.
#' 
#' @param created The date this job was created, in milliseconds from the epoch
#' @param detailedStatus A more detailed description of this job's current status
#' @param errors Any errors that occurred during processing
#' @param id The job ID
#' @param importedIds If this Job represents an import, this field will contain the IDs of the objects that were successfully imported
#' @param projectNumber The Google Developers Console project number to which this job belongs
#' @param request A summarized representation of the original service request
#' @param status The status of this job
#' @param warnings Any warnings that occurred during processing
#' 
#' @return Job object
#' 
#' @family Job functions
#' @export
Job <- function(created = NULL, detailedStatus = NULL, errors = NULL, id = NULL, 
    importedIds = NULL, projectNumber = NULL, request = NULL, status = NULL, warnings = NULL) {
    structure(list(created = created, detailedStatus = detailedStatus, errors = errors, 
        id = id, importedIds = importedIds, projectNumber = projectNumber, request = request, 
        status = status, warnings = warnings), class = "gar_Job")
}

#' JobRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A summary representation of the service request that spawned the job.
#' 
#' @param destination The data destination of the request, for example, a Google BigQuery Table or Dataset ID
#' @param source The data source of the request, for example, a Google Cloud Storage object path or Readset ID
#' @param type The original request type
#' 
#' @return JobRequest object
#' 
#' @family JobRequest functions
#' @export
JobRequest <- function(destination = NULL, source = NULL, type = NULL) {
    structure(list(destination = destination, source = source, type = type), class = "gar_JobRequest")
}

#' KeyValue Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Used to hold basic key value information.
#' 
#' @param key A string which maps to an array of values
#' @param value The string values
#' 
#' @return KeyValue object
#' 
#' @family KeyValue functions
#' @export
KeyValue <- function(key = NULL, value = NULL) {
    structure(list(key = key, value = value), class = "gar_KeyValue")
}

#' LinearAlignment Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A linear alignment can be represented by one CIGAR string. Describes the mapped position and local alignment of the read to the reference.
#' 
#' @param cigar Represents the local alignment of this sequence (alignment matches, indels, etc) against the reference
#' @param mappingQuality The mapping quality of this alignment
#' @param position The position of this alignment
#' 
#' @return LinearAlignment object
#' 
#' @family LinearAlignment functions
#' @export
LinearAlignment <- function(cigar = NULL, mappingQuality = NULL, position = NULL) {
    structure(list(cigar = cigar, mappingQuality = mappingQuality, position = position), 
        class = "gar_LinearAlignment")
}

#' ListBasesResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param offset The offset position (0-based) of the given sequence from the start of this Reference
#' @param sequence A substring of the bases that make up this reference
#' 
#' @return ListBasesResponse object
#' 
#' @family ListBasesResponse functions
#' @export
ListBasesResponse <- function(nextPageToken = NULL, offset = NULL, sequence = NULL) {
    structure(list(nextPageToken = nextPageToken, offset = offset, sequence = sequence), 
        class = "gar_ListBasesResponse")
}

#' ListCoverageBucketsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param bucketWidth The length of each coverage bucket in base pairs
#' @param coverageBuckets The coverage buckets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return ListCoverageBucketsResponse object
#' 
#' @family ListCoverageBucketsResponse functions
#' @export
ListCoverageBucketsResponse <- function(bucketWidth = NULL, coverageBuckets = NULL, 
    nextPageToken = NULL) {
    structure(list(bucketWidth = bucketWidth, coverageBuckets = coverageBuckets, 
        nextPageToken = nextPageToken), class = "gar_ListCoverageBucketsResponse")
}

#' ListDatasetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The dataset list response.
#' 
#' @param datasets The list of matching Datasets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return ListDatasetsResponse object
#' 
#' @family ListDatasetsResponse functions
#' @export
ListDatasetsResponse <- function(datasets = NULL, nextPageToken = NULL) {
    structure(list(datasets = datasets, nextPageToken = nextPageToken), class = "gar_ListDatasetsResponse")
}

#' MergeVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param variants The variants to be merged with existing variants
#' 
#' @return MergeVariantsRequest object
#' 
#' @family MergeVariantsRequest functions
#' @export
MergeVariantsRequest <- function(variants = NULL) {
    structure(list(variants = variants), class = "gar_MergeVariantsRequest")
}

#' Metadata Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' Metadata describes a single piece of variant call metadata. These data include a top level key and either a single value string (value) or a list of key-value pairs (info.) Value and info are mutually exclusive.
#' 
#' @param Metadata.info The \link{Metadata.info} object or list of objects
#' @param description A textual description of this metadata
#' @param id User-provided ID field, not enforced by this API
#' @param info A string which maps to an array of values
#' @param key The top-level key
#' @param number The number of values that can be included in a field described by this metadata
#' @param type The type of data
#' @param value The value field for simple metadata
#' 
#' @return Metadata object
#' 
#' @family Metadata functions
#' @export
Metadata <- function(Metadata.info = NULL, description = NULL, id = NULL, info = NULL, 
    key = NULL, number = NULL, type = NULL, value = NULL) {
    structure(list(Metadata.info = Metadata.info, description = description, id = id, 
        info = info, key = key, number = number, type = type, value = value), class = "gar_Metadata")
}

#' Metadata.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return Metadata.info object
#' 
#' @family Metadata functions
#' @export
Metadata.info <- function() {
    list()
}

#' Position Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' An abstraction for referring to a genomic position, in relation to some already known reference. For now, represents a genomic position as a reference name, a base number on that reference (0-based), and a determination of forward or reverse strand.
#' 
#' @param position The 0-based offset from the start of the forward strand for that reference
#' @param referenceName The name of the reference in whatever reference set is being used
#' @param reverseStrand Whether this position is on the reverse strand, as opposed to the forward strand
#' 
#' @return Position object
#' 
#' @family Position functions
#' @export
Position <- function(position = NULL, referenceName = NULL, reverseStrand = NULL) {
    structure(list(position = position, referenceName = referenceName, reverseStrand = reverseStrand), 
        class = "gar_Position")
}

#' QueryRange Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A 0-based half-open genomic coordinate range for search requests. reference_id or reference_name must be set.
#' 
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param referenceId The ID of the reference to query
#' @param referenceName The name of the reference to query, within the reference set associated with this query
#' @param start The start position of the range on the reference, 0-based inclusive
#' 
#' @return QueryRange object
#' 
#' @family QueryRange functions
#' @export
QueryRange <- function(end = NULL, referenceId = NULL, referenceName = NULL, start = NULL) {
    structure(list(end = end, referenceId = referenceId, referenceName = referenceName, 
        start = start), class = "gar_QueryRange")
}

#' Range Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A 0-based half-open genomic coordinate range over a reference sequence.
#' 
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param referenceName The reference sequence name, for example chr1, 1, or chrX
#' @param start The start position of the range on the reference, 0-based inclusive
#' 
#' @return Range object
#' 
#' @family Range functions
#' @export
Range <- function(end = NULL, referenceName = NULL, start = NULL) {
    structure(list(end = end, referenceName = referenceName, start = start), class = "gar_Range")
}

#' RangePosition Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A 0-based half-open genomic coordinate range over a reference sequence, for representing the position of a genomic resource.
#' 
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param referenceId The ID of the Google Genomics reference associated with this range
#' @param referenceName The display name corresponding to the reference specified by referenceId, for example chr1, 1, or chrX
#' @param reverseStrand Whether this range refers to the reverse strand, as opposed to the forward strand
#' @param start The start position of the range on the reference, 0-based inclusive
#' 
#' @return RangePosition object
#' 
#' @family RangePosition functions
#' @export
RangePosition <- function(end = NULL, referenceId = NULL, referenceName = NULL, reverseStrand = NULL, 
    start = NULL) {
    structure(list(end = end, referenceId = referenceId, referenceName = referenceName, 
        reverseStrand = reverseStrand, start = start), class = "gar_RangePosition")
}

#' Read Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read alignment describes a linear alignment of a string of DNA to a reference sequence, in addition to metadata about the fragment (the molecule of DNA sequenced) and the read (the bases which were read by the sequencer). A read is equivalent to a line in a SAM file. A read belongs to exactly one read group and exactly one read group set. Generating a reference-aligned sequence string When interacting with mapped reads, it's often useful to produce a string representing the local alignment of the read to reference. The following pseudocode demonstrates one way of doing this:out = '' offset = 0 for c in read.alignment.cigar { switch c.operation { case 'ALIGNMENT_MATCH', 'SEQUENCE_MATCH', 'SEQUENCE_MISMATCH': out += read.alignedSequence[offset:offset+c.operationLength] offset += c.operationLength break case 'CLIP_SOFT', 'INSERT': offset += c.operationLength break case 'PAD': out += repeat('*', c.operationLength) break case 'DELETE': out += repeat('-', c.operationLength) break case 'SKIP': out += repeat(' ', c.operationLength) break case 'CLIP_HARD': break } } return outConverting to SAM's CIGAR string The following pseudocode generates a SAM CIGAR string from the cigar field. Note that this is a lossy conversion (cigar.referenceSequence is lost).cigarMap = { 'ALIGNMENT_MATCH': 'M', 'INSERT': 'I', 'DELETE': 'D', 'SKIP': 'N', 'CLIP_SOFT': 'S', 'CLIP_HARD': 'H', 'PAD': 'P', 'SEQUENCE_MATCH': '=', 'SEQUENCE_MISMATCH': 'X', } cigarStr = '' for c in read.alignment.cigar { cigarStr += c.operationLength + cigarMap[c.operation] } return cigarStr
#' 
#' @param Read.info The \link{Read.info} object or list of objects
#' @param alignedQuality The quality of the read sequence contained in this alignment record
#' @param alignedSequence The bases of the read sequence contained in this alignment record, without CIGAR operations applied (equivalent to SEQ in SAM)
#' @param alignment The linear alignment for this alignment record
#' @param duplicateFragment The fragment is a PCR or optical duplicate (SAM flag 0x400)
#' @param failedVendorQualityChecks Whether this read did not pass filters, such as platform or vendor quality controls (SAM flag 0x200)
#' @param fragmentLength The observed length of the fragment, equivalent to TLEN in SAM
#' @param fragmentName The fragment name
#' @param id The unique ID for this read
#' @param info A string which maps to an array of values
#' @param nextMatePosition The position of the primary alignment of the (readNumber+1)%numberReads read in the fragment
#' @param numberReads The number of reads in the fragment (extension to SAM flag 0x1)
#' @param properPlacement The orientation and the distance between reads from the fragment are consistent with the sequencing protocol (SAM flag 0x2)
#' @param readGroupId The ID of the read group this read belongs to
#' @param readGroupSetId The ID of the read group set this read belongs to
#' @param readNumber The read number in sequencing
#' @param secondaryAlignment Whether this alignment is secondary
#' @param supplementaryAlignment Whether this alignment is supplementary
#' 
#' @return Read object
#' 
#' @family Read functions
#' @export
Read <- function(Read.info = NULL, alignedQuality = NULL, alignedSequence = NULL, 
    alignment = NULL, duplicateFragment = NULL, failedVendorQualityChecks = NULL, 
    fragmentLength = NULL, fragmentName = NULL, id = NULL, info = NULL, nextMatePosition = NULL, 
    numberReads = NULL, properPlacement = NULL, readGroupId = NULL, readGroupSetId = NULL, 
    readNumber = NULL, secondaryAlignment = NULL, supplementaryAlignment = NULL) {
    structure(list(Read.info = Read.info, alignedQuality = alignedQuality, alignedSequence = alignedSequence, 
        alignment = alignment, duplicateFragment = duplicateFragment, failedVendorQualityChecks = failedVendorQualityChecks, 
        fragmentLength = fragmentLength, fragmentName = fragmentName, id = id, info = info, 
        nextMatePosition = nextMatePosition, numberReads = numberReads, properPlacement = properPlacement, 
        readGroupId = readGroupId, readGroupSetId = readGroupSetId, readNumber = readNumber, 
        secondaryAlignment = secondaryAlignment, supplementaryAlignment = supplementaryAlignment), 
        class = "gar_Read")
}

#' Read.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return Read.info object
#' 
#' @family Read functions
#' @export
Read.info <- function() {
    list()
}

#' ReadGroup Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read group is all the data that's processed the same way by the sequencer.
#' 
#' @param ReadGroup.info The \link{ReadGroup.info} object or list of objects
#' @param datasetId The ID of the dataset this read group belongs to
#' @param description A free-form text description of this read group
#' @param experiment The experiment used to generate this read group
#' @param id The generated unique read group ID
#' @param info A string which maps to an array of values
#' @param name The read group name
#' @param predictedInsertSize The predicted insert size of this read group
#' @param programs The programs used to generate this read group
#' @param referenceSetId The reference set to which the reads in this read group are aligned
#' @param sampleId The sample this read group's data was generated from
#' 
#' @return ReadGroup object
#' 
#' @family ReadGroup functions
#' @export
ReadGroup <- function(ReadGroup.info = NULL, datasetId = NULL, description = NULL, 
    experiment = NULL, id = NULL, info = NULL, name = NULL, predictedInsertSize = NULL, 
    programs = NULL, referenceSetId = NULL, sampleId = NULL) {
    structure(list(ReadGroup.info = ReadGroup.info, datasetId = datasetId, description = description, 
        experiment = experiment, id = id, info = info, name = name, predictedInsertSize = predictedInsertSize, 
        programs = programs, referenceSetId = referenceSetId, sampleId = sampleId), 
        class = "gar_ReadGroup")
}

#' ReadGroup.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return ReadGroup.info object
#' 
#' @family ReadGroup functions
#' @export
ReadGroup.info <- function() {
    list()
}

#' ReadGroupExperiment Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param instrumentModel The instrument model used as part of this experiment
#' @param libraryId The library used as part of this experiment
#' @param platformUnit The platform unit used as part of this experiment e
#' @param sequencingCenter The sequencing center used as part of this experiment
#' 
#' @return ReadGroupExperiment object
#' 
#' @family ReadGroupExperiment functions
#' @export
ReadGroupExperiment <- function(instrumentModel = NULL, libraryId = NULL, platformUnit = NULL, 
    sequencingCenter = NULL) {
    structure(list(instrumentModel = instrumentModel, libraryId = libraryId, platformUnit = platformUnit, 
        sequencingCenter = sequencingCenter), class = "gar_ReadGroupExperiment")
}

#' ReadGroupProgram Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param commandLine The command line used to run this program
#' @param id The user specified locally unique ID of the program
#' @param name The display name of the program
#' @param prevProgramId The ID of the program run before this one
#' @param version The version of the program run
#' 
#' @return ReadGroupProgram object
#' 
#' @family ReadGroupProgram functions
#' @export
ReadGroupProgram <- function(commandLine = NULL, id = NULL, name = NULL, prevProgramId = NULL, 
    version = NULL) {
    structure(list(commandLine = commandLine, id = id, name = name, prevProgramId = prevProgramId, 
        version = version), class = "gar_ReadGroupProgram")
}

#' ReadGroupSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A read group set is a logical collection of read groups, which are collections of reads produced by a sequencer. A read group set typically models reads corresponding to one sample, sequenced one way, and aligned one way.  - A read group set belongs to one dataset.- A read group belongs to one read group set.- A read belongs to one read group.
#' 
#' @param ReadGroupSet.info The \link{ReadGroupSet.info} object or list of objects
#' @param datasetId The dataset ID
#' @param filename The filename of the original source file for this read group set, if any
#' @param id The read group set ID
#' @param info A string which maps to an array of values
#' @param name The read group set name
#' @param readGroups The read groups in this set
#' @param referenceSetId The reference set the reads in this read group set are aligned to
#' 
#' @return ReadGroupSet object
#' 
#' @family ReadGroupSet functions
#' @export
ReadGroupSet <- function(ReadGroupSet.info = NULL, datasetId = NULL, filename = NULL, 
    id = NULL, info = NULL, name = NULL, readGroups = NULL, referenceSetId = NULL) {
    structure(list(ReadGroupSet.info = ReadGroupSet.info, datasetId = datasetId, 
        filename = filename, id = id, info = info, name = name, readGroups = readGroups, 
        referenceSetId = referenceSetId), class = "gar_ReadGroupSet")
}

#' ReadGroupSet.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return ReadGroupSet.info object
#' 
#' @family ReadGroupSet functions
#' @export
ReadGroupSet.info <- function() {
    list()
}

#' Reference Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A reference is a canonical assembled DNA sequence, intended to act as a reference coordinate space for other genomic annotations. A single reference might represent the human chromosome 1 or mitochandrial DNA, for instance. A reference belongs to one or more reference sets.
#' 
#' @param id The Google generated immutable ID of the reference
#' @param length The length of this reference's sequence
#' @param md5checksum MD5 of the upper-case sequence excluding all whitespace characters (this is equivalent to SQ:M5 in SAM)
#' @param name The name of this reference, for example 22
#' @param ncbiTaxonId ID from http://www
#' @param sourceAccessions All known corresponding accession IDs in INSDC (GenBank/ENA/DDBJ) ideally with a version number, for example GCF_000001405
#' @param sourceURI The URI from which the sequence was obtained
#' 
#' @return Reference object
#' 
#' @family Reference functions
#' @export
Reference <- function(id = NULL, length = NULL, md5checksum = NULL, name = NULL, 
    ncbiTaxonId = NULL, sourceAccessions = NULL, sourceURI = NULL) {
    structure(list(id = id, length = length, md5checksum = md5checksum, name = name, 
        ncbiTaxonId = ncbiTaxonId, sourceAccessions = sourceAccessions, sourceURI = sourceURI), 
        class = "gar_Reference")
}

#' ReferenceBound Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' ReferenceBound records an upper bound for the starting coordinate of variants in a particular reference.
#' 
#' @param referenceName The name of the reference associated with this reference bound
#' @param upperBound An upper bound (inclusive) on the starting coordinate of any variant in the reference sequence
#' 
#' @return ReferenceBound object
#' 
#' @family ReferenceBound functions
#' @export
ReferenceBound <- function(referenceName = NULL, upperBound = NULL) {
    structure(list(referenceName = referenceName, upperBound = upperBound), class = "gar_ReferenceBound")
}

#' ReferenceSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A reference set is a set of references which typically comprise a reference assembly for a species, such as GRCh38 which is representative of the human genome. A reference set defines a common coordinate space for comparing reference-aligned experimental data. A reference set contains 1 or more references.
#' 
#' @param assemblyId Public id of this reference set, such as GRCh37
#' @param description Free text description of this reference set
#' @param id The Google generated immutable ID of the reference set
#' @param md5checksum Order-independent MD5 checksum which identifies this reference set
#' @param ncbiTaxonId ID from http://www
#' @param referenceIds The IDs of the reference objects that are part of this set
#' @param sourceAccessions All known corresponding accession IDs in INSDC (GenBank/ENA/DDBJ) ideally with a version number, for example NC_000001
#' @param sourceURI The URI from which the references were obtained
#' 
#' @return ReferenceSet object
#' 
#' @family ReferenceSet functions
#' @export
ReferenceSet <- function(assemblyId = NULL, description = NULL, id = NULL, md5checksum = NULL, 
    ncbiTaxonId = NULL, referenceIds = NULL, sourceAccessions = NULL, sourceURI = NULL) {
    structure(list(assemblyId = assemblyId, description = description, id = id, md5checksum = md5checksum, 
        ncbiTaxonId = ncbiTaxonId, referenceIds = referenceIds, sourceAccessions = sourceAccessions, 
        sourceURI = sourceURI), class = "gar_ReferenceSet")
}

#' SearchAnnotationSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param datasetIds The dataset IDs to search within
#' @param name Only return annotations sets for which a substring of the name matches this string (case insensitive)
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param referenceSetId If specified, only annotation sets associated with the given reference set are returned
#' @param types If specified, only annotation sets that have any of these types are returned
#' 
#' @return SearchAnnotationSetsRequest object
#' 
#' @family SearchAnnotationSetsRequest functions
#' @export
SearchAnnotationSetsRequest <- function(datasetIds = NULL, name = NULL, pageSize = NULL, 
    pageToken = NULL, referenceSetId = NULL, types = NULL) {
    structure(list(datasetIds = datasetIds, name = name, pageSize = pageSize, pageToken = pageToken, 
        referenceSetId = referenceSetId, types = types), class = "gar_SearchAnnotationSetsRequest")
}

#' SearchAnnotationSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotationSets The matching annotation sets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchAnnotationSetsResponse object
#' 
#' @family SearchAnnotationSetsResponse functions
#' @export
SearchAnnotationSetsResponse <- function(annotationSets = NULL, nextPageToken = NULL) {
    structure(list(annotationSets = annotationSets, nextPageToken = nextPageToken), 
        class = "gar_SearchAnnotationSetsResponse")
}

#' SearchAnnotationsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotationSetIds The annotation sets to search within
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param range If specified, this query matches only annotations that overlap this range
#' 
#' @return SearchAnnotationsRequest object
#' 
#' @family SearchAnnotationsRequest functions
#' @export
SearchAnnotationsRequest <- function(annotationSetIds = NULL, pageSize = NULL, pageToken = NULL, 
    range = NULL) {
    structure(list(annotationSetIds = annotationSetIds, pageSize = pageSize, pageToken = pageToken, 
        range = range), class = "gar_SearchAnnotationsRequest")
}

#' SearchAnnotationsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param annotations The matching annotations
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchAnnotationsResponse object
#' 
#' @family SearchAnnotationsResponse functions
#' @export
SearchAnnotationsResponse <- function(annotations = NULL, nextPageToken = NULL) {
    structure(list(annotations = annotations, nextPageToken = nextPageToken), class = "gar_SearchAnnotationsResponse")
}

#' SearchCallSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The call set search request.
#' 
#' @param name Only return call sets for which a substring of the name matches this string
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param variantSetIds Restrict the query to call sets within the given variant sets
#' 
#' @return SearchCallSetsRequest object
#' 
#' @family SearchCallSetsRequest functions
#' @export
SearchCallSetsRequest <- function(name = NULL, pageSize = NULL, pageToken = NULL, 
    variantSetIds = NULL) {
    structure(list(name = name, pageSize = pageSize, pageToken = pageToken, variantSetIds = variantSetIds), 
        class = "gar_SearchCallSetsRequest")
}

#' SearchCallSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The call set search response.
#' 
#' @param callSets The list of matching call sets
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchCallSetsResponse object
#' 
#' @family SearchCallSetsResponse functions
#' @export
SearchCallSetsResponse <- function(callSets = NULL, nextPageToken = NULL) {
    structure(list(callSets = callSets, nextPageToken = nextPageToken), class = "gar_SearchCallSetsResponse")
}

#' SearchJobsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The jobs search request.
#' 
#' @param createdAfter If specified, only jobs created on or after this date, given in milliseconds since Unix epoch, will be returned
#' @param createdBefore If specified, only jobs created prior to this date, given in milliseconds since Unix epoch, will be returned
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token which is used to page through large result sets
#' @param projectNumber Required
#' @param status Only return jobs which have a matching status
#' 
#' @return SearchJobsRequest object
#' 
#' @family SearchJobsRequest functions
#' @export
SearchJobsRequest <- function(createdAfter = NULL, createdBefore = NULL, pageSize = NULL, 
    pageToken = NULL, projectNumber = NULL, status = NULL) {
    structure(list(createdAfter = createdAfter, createdBefore = createdBefore, pageSize = pageSize, 
        pageToken = pageToken, projectNumber = projectNumber, status = status), class = "gar_SearchJobsRequest")
}

#' SearchJobsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The job search response.
#' 
#' @param jobs The list of jobs results, ordered newest to oldest
#' @param nextPageToken The continuation token which is used to page through large result sets
#' 
#' @return SearchJobsResponse object
#' 
#' @family SearchJobsResponse functions
#' @export
SearchJobsResponse <- function(jobs = NULL, nextPageToken = NULL) {
    structure(list(jobs = jobs, nextPageToken = nextPageToken), class = "gar_SearchJobsResponse")
}

#' SearchReadGroupSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set search request.
#' 
#' @param datasetIds Restricts this query to read group sets within the given datasets
#' @param name Only return read group sets for which a substring of the name matches this string
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReadGroupSetsRequest object
#' 
#' @family SearchReadGroupSetsRequest functions
#' @export
SearchReadGroupSetsRequest <- function(datasetIds = NULL, name = NULL, pageSize = NULL, 
    pageToken = NULL) {
    structure(list(datasetIds = datasetIds, name = name, pageSize = pageSize, pageToken = pageToken), 
        class = "gar_SearchReadGroupSetsRequest")
}

#' SearchReadGroupSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read group set search response.
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param readGroupSets The list of matching read group sets
#' 
#' @return SearchReadGroupSetsResponse object
#' 
#' @family SearchReadGroupSetsResponse functions
#' @export
SearchReadGroupSetsResponse <- function(nextPageToken = NULL, readGroupSets = NULL) {
    structure(list(nextPageToken = nextPageToken, readGroupSets = readGroupSets), 
        class = "gar_SearchReadGroupSetsResponse")
}

#' SearchReadsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read search request.
#' 
#' @param end The end position of the range on the reference, 0-based exclusive
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param readGroupIds The IDs of the read groups within which to search for reads
#' @param readGroupSetIds The IDs of the read groups sets within which to search for reads
#' @param referenceName The reference sequence name, for example chr1, 1, or chrX
#' @param start The start position of the range on the reference, 0-based inclusive
#' 
#' @return SearchReadsRequest object
#' 
#' @family SearchReadsRequest functions
#' @export
SearchReadsRequest <- function(end = NULL, pageSize = NULL, pageToken = NULL, readGroupIds = NULL, 
    readGroupSetIds = NULL, referenceName = NULL, start = NULL) {
    structure(list(end = end, pageSize = pageSize, pageToken = pageToken, readGroupIds = readGroupIds, 
        readGroupSetIds = readGroupSetIds, referenceName = referenceName, start = start), 
        class = "gar_SearchReadsRequest")
}

#' SearchReadsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The read search response.
#' 
#' @param alignments The list of matching alignments sorted by mapped genomic coordinate, if any, ascending in position within the same reference
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReadsResponse object
#' 
#' @family SearchReadsResponse functions
#' @export
SearchReadsResponse <- function(alignments = NULL, nextPageToken = NULL) {
    structure(list(alignments = alignments, nextPageToken = nextPageToken), class = "gar_SearchReadsResponse")
}

#' SearchReferenceSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param accessions If present, return references for which the accession matches any of these strings
#' @param assemblyId If present, return reference sets for which a substring of their assemblyId matches this string (case insensitive)
#' @param md5checksums If present, return references for which the md5checksum matches
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchReferenceSetsRequest object
#' 
#' @family SearchReferenceSetsRequest functions
#' @export
SearchReferenceSetsRequest <- function(accessions = NULL, assemblyId = NULL, md5checksums = NULL, 
    pageSize = NULL, pageToken = NULL) {
    structure(list(accessions = accessions, assemblyId = assemblyId, md5checksums = md5checksums, 
        pageSize = pageSize, pageToken = pageToken), class = "gar_SearchReferenceSetsRequest")
}

#' SearchReferenceSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param referenceSets The matching references sets
#' 
#' @return SearchReferenceSetsResponse object
#' 
#' @family SearchReferenceSetsResponse functions
#' @export
SearchReferenceSetsResponse <- function(nextPageToken = NULL, referenceSets = NULL) {
    structure(list(nextPageToken = nextPageToken, referenceSets = referenceSets), 
        class = "gar_SearchReferenceSetsResponse")
}

#' SearchReferencesRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param accessions If present, return references for which the accession matches this string
#' @param md5checksums If present, return references for which the md5checksum matches
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param referenceSetId If present, return only references which belong to this reference set
#' 
#' @return SearchReferencesRequest object
#' 
#' @family SearchReferencesRequest functions
#' @export
SearchReferencesRequest <- function(accessions = NULL, md5checksums = NULL, pageSize = NULL, 
    pageToken = NULL, referenceSetId = NULL) {
    structure(list(accessions = accessions, md5checksums = md5checksums, pageSize = pageSize, 
        pageToken = pageToken, referenceSetId = referenceSetId), class = "gar_SearchReferencesRequest")
}

#' SearchReferencesResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param references The matching references
#' 
#' @return SearchReferencesResponse object
#' 
#' @family SearchReferencesResponse functions
#' @export
SearchReferencesResponse <- function(nextPageToken = NULL, references = NULL) {
    structure(list(nextPageToken = nextPageToken, references = references), class = "gar_SearchReferencesResponse")
}

#' SearchVariantSetsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The search variant sets request.
#' 
#' @param datasetIds Exactly one dataset ID must be provided here
#' @param pageSize The maximum number of results to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' 
#' @return SearchVariantSetsRequest object
#' 
#' @family SearchVariantSetsRequest functions
#' @export
SearchVariantSetsRequest <- function(datasetIds = NULL, pageSize = NULL, pageToken = NULL) {
    structure(list(datasetIds = datasetIds, pageSize = pageSize, pageToken = pageToken), 
        class = "gar_SearchVariantSetsRequest")
}

#' SearchVariantSetsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The search variant sets response.
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param variantSets The variant sets belonging to the requested dataset
#' 
#' @return SearchVariantSetsResponse object
#' 
#' @family SearchVariantSetsResponse functions
#' @export
SearchVariantSetsResponse <- function(nextPageToken = NULL, variantSets = NULL) {
    structure(list(nextPageToken = nextPageToken, variantSets = variantSets), class = "gar_SearchVariantSetsResponse")
}

#' SearchVariantsRequest Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant search request.
#' 
#' @param callSetIds Only return variant calls which belong to call sets with these ids
#' @param end The end of the window, 0-based exclusive
#' @param maxCalls The maximum number of calls to return in a single page
#' @param pageSize The maximum number of variants to return in a single page
#' @param pageToken The continuation token, which is used to page through large result sets
#' @param referenceName Required
#' @param start The beginning of the window (0-based, inclusive) for which overlapping variants should be returned
#' @param variantName Only return variants which have exactly this name
#' @param variantSetIds At most one variant set ID must be provided
#' 
#' @return SearchVariantsRequest object
#' 
#' @family SearchVariantsRequest functions
#' @export
SearchVariantsRequest <- function(callSetIds = NULL, end = NULL, maxCalls = NULL, 
    pageSize = NULL, pageToken = NULL, referenceName = NULL, start = NULL, variantName = NULL, 
    variantSetIds = NULL) {
    structure(list(callSetIds = callSetIds, end = end, maxCalls = maxCalls, pageSize = pageSize, 
        pageToken = pageToken, referenceName = referenceName, start = start, variantName = variantName, 
        variantSetIds = variantSetIds), class = "gar_SearchVariantsRequest")
}

#' SearchVariantsResponse Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' The variant search response.
#' 
#' @param nextPageToken The continuation token, which is used to page through large result sets
#' @param variants The list of matching Variants
#' 
#' @return SearchVariantsResponse object
#' 
#' @family SearchVariantsResponse functions
#' @export
SearchVariantsResponse <- function(nextPageToken = NULL, variants = NULL) {
    structure(list(nextPageToken = nextPageToken, variants = variants), class = "gar_SearchVariantsResponse")
}

#' Transcript Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A transcript represents the assertion that a particular region of the reference genome may be transcribed as RNA.
#' 
#' @param codingSequence The range of the coding sequence for this transcript, if any
#' @param exons The exons that compose this transcript
#' @param geneId The annotation ID of the gene from which this transcript is transcribed
#' 
#' @return Transcript object
#' 
#' @family Transcript functions
#' @export
Transcript <- function(codingSequence = NULL, exons = NULL, geneId = NULL) {
    structure(list(codingSequence = codingSequence, exons = exons, geneId = geneId), 
        class = "gar_Transcript")
}

#' TranscriptCodingSequence Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param end The end of the coding sequence on this annotation's reference sequence, 0-based exclusive
#' @param start The start of the coding sequence on this annotation's reference sequence, 0-based inclusive
#' 
#' @return TranscriptCodingSequence object
#' 
#' @family TranscriptCodingSequence functions
#' @export
TranscriptCodingSequence <- function(end = NULL, start = NULL) {
    structure(list(end = end, start = start), class = "gar_TranscriptCodingSequence")
}

#' TranscriptExon Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param end The end position of the exon on this annotation's reference sequence, 0-based exclusive
#' @param frame The frame of this exon
#' @param start The start position of the exon on this annotation's reference sequence, 0-based inclusive
#' 
#' @return TranscriptExon object
#' 
#' @family TranscriptExon functions
#' @export
TranscriptExon <- function(end = NULL, frame = NULL, start = NULL) {
    structure(list(end = end, frame = frame, start = start), class = "gar_TranscriptExon")
}

#' Variant Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A variant represents a change in DNA sequence relative to a reference sequence. For example, a variant could represent a SNP or an insertion. Variants belong to a variant set. Each of the calls on a variant represent a determination of genotype with respect to that variant. For example, a call might assign probability of 0.32 to the occurrence of a SNP named rs1234 in a sample named NA12345. A call belongs to a call set, which contains related calls typically from one sample.
#' 
#' @param Variant.info The \link{Variant.info} object or list of objects
#' @param alternateBases The bases that appear instead of the reference bases
#' @param calls The variant calls for this particular variant
#' @param created The date this variant was created, in milliseconds from the epoch
#' @param end The end position (0-based) of this variant
#' @param filter A list of filters (normally quality filters) this variant has failed
#' @param id The Google generated ID of the variant, immutable
#' @param info A string which maps to an array of values
#' @param names Names for the variant, for example a RefSNP ID
#' @param quality A measure of how likely this variant is to be real
#' @param referenceBases The reference bases for this variant
#' @param referenceName The reference on which this variant occurs
#' @param start The position at which this variant occurs (0-based)
#' @param variantSetId The ID of the variant set this variant belongs to
#' 
#' @return Variant object
#' 
#' @family Variant functions
#' @export
Variant <- function(Variant.info = NULL, alternateBases = NULL, calls = NULL, created = NULL, 
    end = NULL, filter = NULL, id = NULL, info = NULL, names = NULL, quality = NULL, 
    referenceBases = NULL, referenceName = NULL, start = NULL, variantSetId = NULL) {
    structure(list(Variant.info = Variant.info, alternateBases = alternateBases, 
        calls = calls, created = created, end = end, filter = filter, id = id, info = info, 
        names = names, quality = quality, referenceBases = referenceBases, referenceName = referenceName, 
        start = start, variantSetId = variantSetId), class = "gar_Variant")
}

#' Variant.info Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A string which maps to an array of values.
#' 
#' 
#' 
#' @return Variant.info object
#' 
#' @family Variant functions
#' @export
Variant.info <- function() {
    list()
}

#' VariantAnnotation Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A Variant annotation.
#' 
#' @param alternateBases The alternate allele for this variant
#' @param clinicalSignificance Describes the clinical significance of a variant
#' @param conditions The set of conditions associated with this variant
#' @param effect Effect of the variant on the coding sequence
#' @param geneId Google annotation ID of the gene affected by this variant
#' @param transcriptIds Google annotation IDs of the transcripts affected by this variant
#' @param type Type has been adapted from ClinVar's list of variant types
#' 
#' @return VariantAnnotation object
#' 
#' @family VariantAnnotation functions
#' @export
VariantAnnotation <- function(alternateBases = NULL, clinicalSignificance = NULL, 
    conditions = NULL, effect = NULL, geneId = NULL, transcriptIds = NULL, type = NULL) {
    structure(list(alternateBases = alternateBases, clinicalSignificance = clinicalSignificance, 
        conditions = conditions, effect = effect, geneId = geneId, transcriptIds = transcriptIds, 
        type = type), class = "gar_VariantAnnotation")
}

#' VariantAnnotationCondition Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' No description
#' 
#' @param conceptId The MedGen concept id associated with this gene
#' @param externalIds The set of external IDs for this condition
#' @param names A set of names for the condition
#' @param omimId The OMIM id for this condition
#' 
#' @return VariantAnnotationCondition object
#' 
#' @family VariantAnnotationCondition functions
#' @export
VariantAnnotationCondition <- function(conceptId = NULL, externalIds = NULL, names = NULL, 
    omimId = NULL) {
    structure(list(conceptId = conceptId, externalIds = externalIds, names = names, 
        omimId = omimId), class = "gar_VariantAnnotationCondition")
}


#' VariantSet Object
#' 
#' @details 
#' Autogenerated via \code{\link[googleAuthR]{gar_create_api_objects}}
#' A variant set is a collection of call sets and variants. It contains summary statistics of those contents. A variant set belongs to a dataset.
#' 
#' @param datasetId The dataset to which this variant set belongs
#' @param description A textual description of this variant set
#' @param id The Google-generated ID of the variant set
#' @param metadata The metadata associated with this variant set
#' @param name User-specified, mutable name
#' @param referenceBounds A list of all references used by the variants in a variant set with associated coordinate upper bounds for each one
#' @param referenceSetId The reference set to which the variant set is mapped
#' 
#' @return VariantSet object
#' 
#' @family VariantSet functions
#' @export


VariantSet <- function(datasetId = NULL, description = NULL, id = NULL, metadata = NULL, 
    name = NULL, referenceBounds = NULL, referenceSetId = NULL) {
    
    
    
    structure(list(datasetId = datasetId, description = description, id = id, metadata = metadata, 
        name = name, referenceBounds = referenceBounds, referenceSetId = referenceSetId), 
        class = "gar_VariantSet")
}

